{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "667ccd5a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd, numpy as np\n",
    "\n",
    "pd.set_option('display.max_rows', None)\n",
    "pd.set_option('display.max_columns', None)\n",
    "pd.set_option('display.expand_frame_repr', False)\n",
    "pd.set_option('display.max_colwidth', None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4960e0a3",
   "metadata": {},
   "source": [
    "### ClinVar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "30998280",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/cc/jxs8yxmn00bb_43d9z_0f13c0000gn/T/ipykernel_48435/2626421186.py:1: DtypeWarning: Columns (72) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  clinvar_allpreds = pd.read_csv('~/Desktop/analysis files/cleaned_ClinVar_ALL_preds.csv', sep=',')\n"
     ]
    }
   ],
   "source": [
    "clinvar_allpreds = pd.read_csv('~/Desktop/analysis files/cleaned_ClinVar_ALL_preds.csv', sep=',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "738783cd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(46050, 74)"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_allpreds.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "7e5c5569",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#AlleleID</th>\n",
       "      <th>Type</th>\n",
       "      <th>Name</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>nsv/esv (dbVar)</th>\n",
       "      <th>RCVaccession</th>\n",
       "      <th>PhenotypeIDS</th>\n",
       "      <th>PhenotypeList</th>\n",
       "      <th>Origin</th>\n",
       "      <th>OriginSimple</th>\n",
       "      <th>Assembly</th>\n",
       "      <th>ChromosomeAccession</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start</th>\n",
       "      <th>Stop</th>\n",
       "      <th>Cytogenetic</th>\n",
       "      <th>ReviewStatus</th>\n",
       "      <th>NumberSubmitters</th>\n",
       "      <th>OtherIDs</th>\n",
       "      <th>SubmitterCategories</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PositionVCF</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "      <th>Variant (3-letter)</th>\n",
       "      <th>Variant</th>\n",
       "      <th>LastEvaluated (Year)</th>\n",
       "      <th>Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>Extra</th>\n",
       "      <th>PositionVCF_dashed</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomAD_AF</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>BinaryClinicalSignificance</th>\n",
       "      <th>SIFT_label</th>\n",
       "      <th>SIFT_score</th>\n",
       "      <th>PolyPhen_label</th>\n",
       "      <th>PolyPhen_score</th>\n",
       "      <th>BayesDel_label</th>\n",
       "      <th>BayesDel_score</th>\n",
       "      <th>CADD_PHRED_score</th>\n",
       "      <th>CADD_RAW_score</th>\n",
       "      <th>ClinPred_score</th>\n",
       "      <th>VEST4_score</th>\n",
       "      <th>VEST4_rankscore</th>\n",
       "      <th>EVE_label</th>\n",
       "      <th>EVE_score</th>\n",
       "      <th>REVEL_score</th>\n",
       "      <th>PrimateAI_score</th>\n",
       "      <th>AM_label</th>\n",
       "      <th>AM_score</th>\n",
       "      <th>UniprotID</th>\n",
       "      <th>Envision_predictions</th>\n",
       "      <th>QAFI_predictions</th>\n",
       "      <th>MetaRNN_score</th>\n",
       "      <th>MetaRNN_pred</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>317931</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_015665.6(AAAS):c.1597G&gt;A (p.Gly533Arg)</td>\n",
       "      <td>8086</td>\n",
       "      <td>AAAS</td>\n",
       "      <td>HGNC:13666</td>\n",
       "      <td>Benign/Likely benign</td>\n",
       "      <td>0</td>\n",
       "      <td>34451260</td>\n",
       "      <td>-</td>\n",
       "      <td>RCV000343022|RCV000886859</td>\n",
       "      <td>MONDO:MONDO:0009279,MedGen:C0271742,OMIM:231550,Orphanet:869|MedGen:C3661900</td>\n",
       "      <td>Glucocorticoid deficiency with achalasia|not provided</td>\n",
       "      <td>germline</td>\n",
       "      <td>germline</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>12</td>\n",
       "      <td>53307533</td>\n",
       "      <td>53307533</td>\n",
       "      <td>12q13.13</td>\n",
       "      <td>criteria provided, multiple submitters, no conflicts</td>\n",
       "      <td>3</td>\n",
       "      <td>ClinGen:CA6598796</td>\n",
       "      <td>2</td>\n",
       "      <td>309718</td>\n",
       "      <td>53307533</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Gly533Arg</td>\n",
       "      <td>G533R</td>\n",
       "      <td>2024</td>\n",
       "      <td>3.445126e+07</td>\n",
       "      <td>12:53307533</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000094914</td>\n",
       "      <td>ENST00000209873.9</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1742</td>\n",
       "      <td>1597</td>\n",
       "      <td>533</td>\n",
       "      <td>G/R</td>\n",
       "      <td>Ggg/Agg</td>\n",
       "      <td>rs34451260</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;SYMBOL=AAAS;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:13666;CCDS=CCDS8856.1;ENSP=ENSP00000209873;SWISSPROT=Q9NRG9.206;UNIPARC=UPI0000039E40;UNIPROT_ISOFORM=Q9NRG9-1;HGVSc=ENST00000209873.9:c.1597G&gt;A;HGVSp=ENSP00000209873.4:p.Gly533Arg;AF=0.0124;AFR_AF=0.0439;AMR_AF=0.0043;EAS_AF=0;EUR_AF=0.001;SAS_AF=0;gnomADe_AF=0.0007928;gnomADe_AFR_AF=0.02712;gnomADe_AMR_AF=0.001879;gnomADe_ASJ_AF=3.827e-05;gnomADe_EAS_AF=0;gnomADe_FIN_AF=0;gnomADe_MID_AF=0.001214;gnomADe_NFE_AF=2.428e-05;gnomADe_REMAINING_AF=0.002053;gnomADe_SAS_AF=9.275e-05;gnomADg_AF=0.007867;gnomADg_AFR_AF=0.02713;gnomADg_AMI_AF=0;gnomADg_AMR_AF=0.003529;gnomADg_ASJ_AF=0;gnomADg_EAS_AF=0;gnomADg_FIN_AF=0;gnomADg_MID_AF=0;gnomADg_NFE_AF=0.0001029;gnomADg_REMAINING_AF=0.004748;gnomADg_SAS_AF=0;MAX_AF=0.0439;MAX_AF_POPS=AFR;CLIN_SIG=benign,likely_benign;PHENO=1</td>\n",
       "      <td>53307533</td>\n",
       "      <td>0.000793</td>\n",
       "      <td>0.007867</td>\n",
       "      <td>0.000793</td>\n",
       "      <td>ENSP00000209873.4:p.Gly533Arg</td>\n",
       "      <td>B</td>\n",
       "      <td>deleterious_low_confidence</td>\n",
       "      <td>0.00</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.362112</td>\n",
       "      <td>20.9</td>\n",
       "      <td>2.951810</td>\n",
       "      <td>0.062144</td>\n",
       "      <td>0.420</td>\n",
       "      <td>0.45743</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.257</td>\n",
       "      <td>0.526710</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.1192</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>0.977286</td>\n",
       "      <td>0.852</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20084</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_015665.6(AAAS):c.787T&gt;C (p.Ser263Pro)</td>\n",
       "      <td>8086</td>\n",
       "      <td>AAAS</td>\n",
       "      <td>HGNC:13666</td>\n",
       "      <td>Pathogenic/Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>121918550</td>\n",
       "      <td>-</td>\n",
       "      <td>RCV000005348|RCV000311283|RCV000415076|RCV000624696</td>\n",
       "      <td>MONDO:MONDO:0009279,MedGen:C0271742,OMIM:231550,Orphanet:869|MedGen:C3661900|Human Phenotype Ontology:HP:0002313,Human Phenotype Ontology:HP:0007191,MedGen:C0037771;Human Phenotype Ontology:HP:0001282,Human Phenotype Ontology:HP:0001347,Human Phenotype Ontology:HP:0006820,Human Phenotype Ontology:HP:0007184,Human Phenotype Ontology:HP:0007318,MONDO:MONDO:0007774,MedGen:C0151889,OMIM:145290;Human Phenotype Ontology:HP:0001352,Human Phenotype Ontology:HP:0003487,MedGen:C0034935|MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>Glucocorticoid deficiency with achalasia|not provided|Spastic paraparesis;Hyperreflexia;Babinski sign|Inborn genetic diseases</td>\n",
       "      <td>germline;unknown</td>\n",
       "      <td>germline</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>12</td>\n",
       "      <td>53309624</td>\n",
       "      <td>53309624</td>\n",
       "      <td>12q13.13</td>\n",
       "      <td>criteria provided, multiple submitters, no conflicts</td>\n",
       "      <td>11</td>\n",
       "      <td>ClinGen:CA117228,UniProtKB:Q9NRG9#VAR_012806,OMIM:605378.0007</td>\n",
       "      <td>3</td>\n",
       "      <td>5045</td>\n",
       "      <td>53309624</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Ser263Pro</td>\n",
       "      <td>S263P</td>\n",
       "      <td>2024</td>\n",
       "      <td>1.219186e+08</td>\n",
       "      <td>12:53309624</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000094914</td>\n",
       "      <td>ENST00000209873.9</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>932</td>\n",
       "      <td>787</td>\n",
       "      <td>263</td>\n",
       "      <td>S/P</td>\n",
       "      <td>Tca/Cca</td>\n",
       "      <td>rs121918550,CM010150</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;SYMBOL=AAAS;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:13666;CCDS=CCDS8856.1;ENSP=ENSP00000209873;SWISSPROT=Q9NRG9.206;UNIPARC=UPI0000039E40;UNIPROT_ISOFORM=Q9NRG9-1;HGVSc=ENST00000209873.9:c.787T&gt;C;HGVSp=ENSP00000209873.4:p.Ser263Pro;AF=0.0002;AFR_AF=0;AMR_AF=0;EAS_AF=0;EUR_AF=0.001;SAS_AF=0;gnomADe_AF=7.801e-05;gnomADe_AFR_AF=2.988e-05;gnomADe_AMR_AF=0;gnomADe_ASJ_AF=0;gnomADe_EAS_AF=0;gnomADe_FIN_AF=0.0005994;gnomADe_MID_AF=0;gnomADe_NFE_AF=6.925e-05;gnomADe_REMAINING_AF=6.627e-05;gnomADe_SAS_AF=0;gnomADg_AF=5.908e-05;gnomADg_AFR_AF=0;gnomADg_AMI_AF=0;gnomADg_AMR_AF=0;gnomADg_ASJ_AF=0;gnomADg_EAS_AF=0;gnomADg_FIN_AF=0.0002824;gnomADg_MID_AF=0;gnomADg_NFE_AF=8.819e-05;gnomADg_REMAINING_AF=0;gnomADg_SAS_AF=0;MAX_AF=0.001;MAX_AF_POPS=EUR;CLIN_SIG=pathogenic/likely_pathogenic,pathogenic;SOMATIC=0,1;PHENO=1,1</td>\n",
       "      <td>53309624</td>\n",
       "      <td>0.000078</td>\n",
       "      <td>0.000059</td>\n",
       "      <td>0.000078</td>\n",
       "      <td>ENSP00000209873.4:p.Ser263Pro</td>\n",
       "      <td>P</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>0.00</td>\n",
       "      <td>probably_damaging</td>\n",
       "      <td>0.992</td>\n",
       "      <td>D</td>\n",
       "      <td>0.258310</td>\n",
       "      <td>29.2</td>\n",
       "      <td>5.211912</td>\n",
       "      <td>0.980945</td>\n",
       "      <td>0.983</td>\n",
       "      <td>0.99270</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>0.951397</td>\n",
       "      <td>0.844</td>\n",
       "      <td>0.718980</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.9852</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>0.860283</td>\n",
       "      <td>0.458</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1312858</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_015665.6(AAAS):c.500C&gt;T (p.Ala167Val)</td>\n",
       "      <td>8086</td>\n",
       "      <td>AAAS</td>\n",
       "      <td>HGNC:13666</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>1017700992</td>\n",
       "      <td>-</td>\n",
       "      <td>RCV003557743|RCV004765890</td>\n",
       "      <td>MedGen:C3661900|MONDO:MONDO:0009279,MedGen:C0271742,OMIM:231550,Orphanet:869</td>\n",
       "      <td>not provided|Glucocorticoid deficiency with achalasia</td>\n",
       "      <td>germline</td>\n",
       "      <td>germline</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>12</td>\n",
       "      <td>53314796</td>\n",
       "      <td>53314796</td>\n",
       "      <td>12q13.13</td>\n",
       "      <td>criteria provided, multiple submitters, no conflicts</td>\n",
       "      <td>2</td>\n",
       "      <td>ClinGen:CA237335767</td>\n",
       "      <td>2</td>\n",
       "      <td>2735891</td>\n",
       "      <td>53314796</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Ala167Val</td>\n",
       "      <td>A167V</td>\n",
       "      <td>2024</td>\n",
       "      <td>1.017701e+09</td>\n",
       "      <td>12:53314796</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000094914</td>\n",
       "      <td>ENST00000209873.9</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>645</td>\n",
       "      <td>500</td>\n",
       "      <td>167</td>\n",
       "      <td>A/V</td>\n",
       "      <td>gCa/gTa</td>\n",
       "      <td>rs1017700992,CM065949</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;SYMBOL=AAAS;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:13666;CCDS=CCDS8856.1;ENSP=ENSP00000209873;SWISSPROT=Q9NRG9.206;UNIPARC=UPI0000039E40;UNIPROT_ISOFORM=Q9NRG9-1;HGVSc=ENST00000209873.9:c.500C&gt;T;HGVSp=ENSP00000209873.4:p.Ala167Val;gnomADe_AF=3.421e-06;gnomADe_AFR_AF=0;gnomADe_AMR_AF=0;gnomADe_ASJ_AF=0;gnomADe_EAS_AF=0;gnomADe_FIN_AF=0;gnomADe_MID_AF=0;gnomADe_NFE_AF=4.497e-06;gnomADe_REMAINING_AF=0;gnomADe_SAS_AF=0;MAX_AF=4.497e-06;MAX_AF_POPS=gnomADe_NFE;CLIN_SIG=pathogenic;SOMATIC=0,1;PHENO=1,1</td>\n",
       "      <td>53314796</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>ENSP00000209873.4:p.Ala167Val</td>\n",
       "      <td>P</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>0.03</td>\n",
       "      <td>probably_damaging</td>\n",
       "      <td>0.969</td>\n",
       "      <td>D</td>\n",
       "      <td>0.571933</td>\n",
       "      <td>34.0</td>\n",
       "      <td>5.972388</td>\n",
       "      <td>0.998261</td>\n",
       "      <td>0.980</td>\n",
       "      <td>0.99026</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>0.853707</td>\n",
       "      <td>0.920</td>\n",
       "      <td>0.760144</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.8614</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>0.918544</td>\n",
       "      <td>0.766</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   #AlleleID                       Type                                       Name  GeneID GeneSymbol     HGNC_ID          ClinicalSignificance  ClinSigSimple  RS# (dbSNP) nsv/esv (dbVar)                                         RCVaccession                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   PhenotypeIDS                                                                                                                  PhenotypeList            Origin OriginSimple Assembly ChromosomeAccession Chromosome     Start      Stop Cytogenetic                                          ReviewStatus  NumberSubmitters                                                       OtherIDs  SubmitterCategories  VariationID  PositionVCF ReferenceAlleleVCF AlternateAlleleVCF Variant (3-letter) Variant  LastEvaluated (Year)  Uploaded_variation     Location Allele             Gene            Feature Feature_type       Consequence  cDNA_position  CDS_position  Protein_position Amino_acids   Codons     Existing_variation                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Extra  PositionVCF_dashed  gnomADe_AF  gnomADg_AF  gnomAD_AF                          HGVSp BinaryClinicalSignificance                  SIFT_label  SIFT_score     PolyPhen_label  PolyPhen_score BayesDel_label  BayesDel_score  CADD_PHRED_score  CADD_RAW_score  ClinPred_score  VEST4_score  VEST4_rankscore   EVE_label  EVE_score  REVEL_score  PrimateAI_score           AM_label  AM_score UniprotID  Envision_predictions  QAFI_predictions MetaRNN_score MetaRNN_pred\n",
       "0     317931  single nucleotide variant  NM_015665.6(AAAS):c.1597G>A (p.Gly533Arg)    8086       AAAS  HGNC:13666          Benign/Likely benign              0     34451260               -                            RCV000343022|RCV000886859                                                                                                                                                                                                                                                                                                                                                                                                                                                   MONDO:MONDO:0009279,MedGen:C0271742,OMIM:231550,Orphanet:869|MedGen:C3661900                                                                          Glucocorticoid deficiency with achalasia|not provided          germline     germline   GRCh38        NC_000012.12         12  53307533  53307533    12q13.13  criteria provided, multiple submitters, no conflicts                 3                                              ClinGen:CA6598796                    2       309718     53307533                  C                  T          Gly533Arg   G533R                  2024        3.445126e+07  12:53307533      T  ENSG00000094914  ENST00000209873.9   Transcript  missense_variant           1742          1597               533         G/R  Ggg/Agg             rs34451260  IMPACT=MODERATE;STRAND=-1;SYMBOL=AAAS;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:13666;CCDS=CCDS8856.1;ENSP=ENSP00000209873;SWISSPROT=Q9NRG9.206;UNIPARC=UPI0000039E40;UNIPROT_ISOFORM=Q9NRG9-1;HGVSc=ENST00000209873.9:c.1597G>A;HGVSp=ENSP00000209873.4:p.Gly533Arg;AF=0.0124;AFR_AF=0.0439;AMR_AF=0.0043;EAS_AF=0;EUR_AF=0.001;SAS_AF=0;gnomADe_AF=0.0007928;gnomADe_AFR_AF=0.02712;gnomADe_AMR_AF=0.001879;gnomADe_ASJ_AF=3.827e-05;gnomADe_EAS_AF=0;gnomADe_FIN_AF=0;gnomADe_MID_AF=0.001214;gnomADe_NFE_AF=2.428e-05;gnomADe_REMAINING_AF=0.002053;gnomADe_SAS_AF=9.275e-05;gnomADg_AF=0.007867;gnomADg_AFR_AF=0.02713;gnomADg_AMI_AF=0;gnomADg_AMR_AF=0.003529;gnomADg_ASJ_AF=0;gnomADg_EAS_AF=0;gnomADg_FIN_AF=0;gnomADg_MID_AF=0;gnomADg_NFE_AF=0.0001029;gnomADg_REMAINING_AF=0.004748;gnomADg_SAS_AF=0;MAX_AF=0.0439;MAX_AF_POPS=AFR;CLIN_SIG=benign,likely_benign;PHENO=1            53307533    0.000793    0.007867   0.000793  ENSP00000209873.4:p.Gly533Arg                          B  deleterious_low_confidence        0.00            unknown           0.000              T       -0.362112              20.9        2.951810        0.062144        0.420          0.45743         NaN        NaN        0.257         0.526710      likely_benign    0.1192    Q9NRG9              0.977286             0.852           NaN          NaN\n",
       "1      20084  single nucleotide variant   NM_015665.6(AAAS):c.787T>C (p.Ser263Pro)    8086       AAAS  HGNC:13666  Pathogenic/Likely pathogenic              1    121918550               -  RCV000005348|RCV000311283|RCV000415076|RCV000624696  MONDO:MONDO:0009279,MedGen:C0271742,OMIM:231550,Orphanet:869|MedGen:C3661900|Human Phenotype Ontology:HP:0002313,Human Phenotype Ontology:HP:0007191,MedGen:C0037771;Human Phenotype Ontology:HP:0001282,Human Phenotype Ontology:HP:0001347,Human Phenotype Ontology:HP:0006820,Human Phenotype Ontology:HP:0007184,Human Phenotype Ontology:HP:0007318,MONDO:MONDO:0007774,MedGen:C0151889,OMIM:145290;Human Phenotype Ontology:HP:0001352,Human Phenotype Ontology:HP:0003487,MedGen:C0034935|MeSH:D030342,MedGen:C0950123  Glucocorticoid deficiency with achalasia|not provided|Spastic paraparesis;Hyperreflexia;Babinski sign|Inborn genetic diseases  germline;unknown     germline   GRCh38        NC_000012.12         12  53309624  53309624    12q13.13  criteria provided, multiple submitters, no conflicts                11  ClinGen:CA117228,UniProtKB:Q9NRG9#VAR_012806,OMIM:605378.0007                    3         5045     53309624                  A                  G          Ser263Pro   S263P                  2024        1.219186e+08  12:53309624      G  ENSG00000094914  ENST00000209873.9   Transcript  missense_variant            932           787               263         S/P  Tca/Cca   rs121918550,CM010150           IMPACT=MODERATE;STRAND=-1;SYMBOL=AAAS;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:13666;CCDS=CCDS8856.1;ENSP=ENSP00000209873;SWISSPROT=Q9NRG9.206;UNIPARC=UPI0000039E40;UNIPROT_ISOFORM=Q9NRG9-1;HGVSc=ENST00000209873.9:c.787T>C;HGVSp=ENSP00000209873.4:p.Ser263Pro;AF=0.0002;AFR_AF=0;AMR_AF=0;EAS_AF=0;EUR_AF=0.001;SAS_AF=0;gnomADe_AF=7.801e-05;gnomADe_AFR_AF=2.988e-05;gnomADe_AMR_AF=0;gnomADe_ASJ_AF=0;gnomADe_EAS_AF=0;gnomADe_FIN_AF=0.0005994;gnomADe_MID_AF=0;gnomADe_NFE_AF=6.925e-05;gnomADe_REMAINING_AF=6.627e-05;gnomADe_SAS_AF=0;gnomADg_AF=5.908e-05;gnomADg_AFR_AF=0;gnomADg_AMI_AF=0;gnomADg_AMR_AF=0;gnomADg_ASJ_AF=0;gnomADg_EAS_AF=0;gnomADg_FIN_AF=0.0002824;gnomADg_MID_AF=0;gnomADg_NFE_AF=8.819e-05;gnomADg_REMAINING_AF=0;gnomADg_SAS_AF=0;MAX_AF=0.001;MAX_AF_POPS=EUR;CLIN_SIG=pathogenic/likely_pathogenic,pathogenic;SOMATIC=0,1;PHENO=1,1            53309624    0.000078    0.000059   0.000078  ENSP00000209873.4:p.Ser263Pro                          P                 deleterious        0.00  probably_damaging           0.992              D        0.258310              29.2        5.211912        0.980945        0.983          0.99270  Pathogenic   0.951397        0.844         0.718980  likely_pathogenic    0.9852    Q9NRG9              0.860283             0.458           NaN          NaN\n",
       "2    1312858  single nucleotide variant   NM_015665.6(AAAS):c.500C>T (p.Ala167Val)    8086       AAAS  HGNC:13666                    Pathogenic              1   1017700992               -                            RCV003557743|RCV004765890                                                                                                                                                                                                                                                                                                                                                                                                                                                   MedGen:C3661900|MONDO:MONDO:0009279,MedGen:C0271742,OMIM:231550,Orphanet:869                                                                          not provided|Glucocorticoid deficiency with achalasia          germline     germline   GRCh38        NC_000012.12         12  53314796  53314796    12q13.13  criteria provided, multiple submitters, no conflicts                 2                                            ClinGen:CA237335767                    2      2735891     53314796                  G                  A          Ala167Val   A167V                  2024        1.017701e+09  12:53314796      A  ENSG00000094914  ENST00000209873.9   Transcript  missense_variant            645           500               167         A/V  gCa/gTa  rs1017700992,CM065949                                                                                                                                                                                                                                                                                                                                    IMPACT=MODERATE;STRAND=-1;SYMBOL=AAAS;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:13666;CCDS=CCDS8856.1;ENSP=ENSP00000209873;SWISSPROT=Q9NRG9.206;UNIPARC=UPI0000039E40;UNIPROT_ISOFORM=Q9NRG9-1;HGVSc=ENST00000209873.9:c.500C>T;HGVSp=ENSP00000209873.4:p.Ala167Val;gnomADe_AF=3.421e-06;gnomADe_AFR_AF=0;gnomADe_AMR_AF=0;gnomADe_ASJ_AF=0;gnomADe_EAS_AF=0;gnomADe_FIN_AF=0;gnomADe_MID_AF=0;gnomADe_NFE_AF=4.497e-06;gnomADe_REMAINING_AF=0;gnomADe_SAS_AF=0;MAX_AF=4.497e-06;MAX_AF_POPS=gnomADe_NFE;CLIN_SIG=pathogenic;SOMATIC=0,1;PHENO=1,1            53314796    0.000003         NaN   0.000003  ENSP00000209873.4:p.Ala167Val                          P                 deleterious        0.03  probably_damaging           0.969              D        0.571933              34.0        5.972388        0.998261        0.980          0.99026  Pathogenic   0.853707        0.920         0.760144  likely_pathogenic    0.8614    Q9NRG9              0.918544             0.766           NaN          NaN"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_allpreds.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "20fb795f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "BayesDel_label             55\n",
       "BayesDel_score             55\n",
       "CADD_PHRED_score            0\n",
       "CADD_RAW_score              0\n",
       "VEST4_score                 0\n",
       "EVE_label               15760\n",
       "EVE_score               15760\n",
       "REVEL_score              3876\n",
       "AM_label                 2879\n",
       "AM_score                 2879\n",
       "Envision_predictions     3337\n",
       "QAFI_predictions        16427\n",
       "MetaRNN_score           36563\n",
       "MetaRNN_pred            36563\n",
       "dtype: int64"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_allpreds[['BayesDel_label', 'BayesDel_score', 'CADD_PHRED_score', 'CADD_RAW_score', 'VEST4_score', 'EVE_label', 'EVE_score', 'REVEL_score', 'AM_label', 'AM_score', 'Envision_predictions', 'QAFI_predictions', 'MetaRNN_score', 'MetaRNN_pred']].isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "469fd6d7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SIFT_label: ['deleterious_low_confidence' 'deleterious' 'tolerated'\n",
      " 'tolerated_low_confidence' nan]\n",
      "PolyPhen_label: ['unknown' 'probably_damaging' 'benign' 'possibly_damaging' nan]\n",
      "BayesDel_label: ['T' 'D' nan]\n",
      "EVE_label: [nan 'Pathogenic' 'Benign' 'Uncertain']\n",
      "AM_label: ['likely_benign' 'likely_pathogenic' 'ambiguous' nan]\n",
      "MetaRNN_pred: [nan 'D' 'T' '.']\n"
     ]
    }
   ],
   "source": [
    "for col in clinvar_allpreds.columns:\n",
    "    if col.endswith('label') or col.endswith('pred'):\n",
    "        print(f\"{col}: {clinvar_allpreds[col].unique()}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "ed5f26dd",
   "metadata": {},
   "outputs": [],
   "source": [
    "mappings = {\n",
    "    'SIFT_label': {'deleterious': 1, 'deleterious_low_confidence': 1, 'tolerated': 0, 'tolerated_low_confidence': 0},\n",
    "    'PolyPhen_label': {'probably_damaging': 1, 'possibly_damaging': 1, 'benign': 0},\n",
    "    'BayesDel_label': {'T': 0, 'D': 1},\n",
    "    'EVE_label': {'Benign': 0, 'Pathogenic': 1},\n",
    "    'AM_label': {'likely_benign': 0, 'likely_pathogenic': 1},\n",
    "    'MetaRNN_pred': {'T': 0, 'D': 1}\n",
    "}\n",
    "\n",
    "for col, mapping in mappings.items():\n",
    "    clinvar_allpreds[col] = clinvar_allpreds[col].map(mapping).astype('Int64')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "946707b2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#AlleleID</th>\n",
       "      <th>Type</th>\n",
       "      <th>Name</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>nsv/esv (dbVar)</th>\n",
       "      <th>RCVaccession</th>\n",
       "      <th>PhenotypeIDS</th>\n",
       "      <th>PhenotypeList</th>\n",
       "      <th>Origin</th>\n",
       "      <th>OriginSimple</th>\n",
       "      <th>Assembly</th>\n",
       "      <th>ChromosomeAccession</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start</th>\n",
       "      <th>Stop</th>\n",
       "      <th>Cytogenetic</th>\n",
       "      <th>ReviewStatus</th>\n",
       "      <th>NumberSubmitters</th>\n",
       "      <th>OtherIDs</th>\n",
       "      <th>SubmitterCategories</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PositionVCF</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "      <th>Variant (3-letter)</th>\n",
       "      <th>Variant</th>\n",
       "      <th>LastEvaluated (Year)</th>\n",
       "      <th>Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>Extra</th>\n",
       "      <th>PositionVCF_dashed</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomAD_AF</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>BinaryClinicalSignificance</th>\n",
       "      <th>SIFT_label</th>\n",
       "      <th>SIFT_score</th>\n",
       "      <th>PolyPhen_label</th>\n",
       "      <th>PolyPhen_score</th>\n",
       "      <th>BayesDel_label</th>\n",
       "      <th>BayesDel_score</th>\n",
       "      <th>CADD_PHRED_score</th>\n",
       "      <th>CADD_RAW_score</th>\n",
       "      <th>ClinPred_score</th>\n",
       "      <th>VEST4_score</th>\n",
       "      <th>VEST4_rankscore</th>\n",
       "      <th>EVE_label</th>\n",
       "      <th>EVE_score</th>\n",
       "      <th>REVEL_score</th>\n",
       "      <th>PrimateAI_score</th>\n",
       "      <th>AM_label</th>\n",
       "      <th>AM_score</th>\n",
       "      <th>UniprotID</th>\n",
       "      <th>Envision_predictions</th>\n",
       "      <th>QAFI_predictions</th>\n",
       "      <th>MetaRNN_score</th>\n",
       "      <th>MetaRNN_pred</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>317931</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_015665.6(AAAS):c.1597G&gt;A (p.Gly533Arg)</td>\n",
       "      <td>8086</td>\n",
       "      <td>AAAS</td>\n",
       "      <td>HGNC:13666</td>\n",
       "      <td>Benign/Likely benign</td>\n",
       "      <td>0</td>\n",
       "      <td>34451260</td>\n",
       "      <td>-</td>\n",
       "      <td>RCV000343022|RCV000886859</td>\n",
       "      <td>MONDO:MONDO:0009279,MedGen:C0271742,OMIM:231550,Orphanet:869|MedGen:C3661900</td>\n",
       "      <td>Glucocorticoid deficiency with achalasia|not provided</td>\n",
       "      <td>germline</td>\n",
       "      <td>germline</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>12</td>\n",
       "      <td>53307533</td>\n",
       "      <td>53307533</td>\n",
       "      <td>12q13.13</td>\n",
       "      <td>criteria provided, multiple submitters, no conflicts</td>\n",
       "      <td>3</td>\n",
       "      <td>ClinGen:CA6598796</td>\n",
       "      <td>2</td>\n",
       "      <td>309718</td>\n",
       "      <td>53307533</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Gly533Arg</td>\n",
       "      <td>G533R</td>\n",
       "      <td>2024</td>\n",
       "      <td>3.445126e+07</td>\n",
       "      <td>12:53307533</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000094914</td>\n",
       "      <td>ENST00000209873.9</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1742</td>\n",
       "      <td>1597</td>\n",
       "      <td>533</td>\n",
       "      <td>G/R</td>\n",
       "      <td>Ggg/Agg</td>\n",
       "      <td>rs34451260</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;SYMBOL=AAAS;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:13666;CCDS=CCDS8856.1;ENSP=ENSP00000209873;SWISSPROT=Q9NRG9.206;UNIPARC=UPI0000039E40;UNIPROT_ISOFORM=Q9NRG9-1;HGVSc=ENST00000209873.9:c.1597G&gt;A;HGVSp=ENSP00000209873.4:p.Gly533Arg;AF=0.0124;AFR_AF=0.0439;AMR_AF=0.0043;EAS_AF=0;EUR_AF=0.001;SAS_AF=0;gnomADe_AF=0.0007928;gnomADe_AFR_AF=0.02712;gnomADe_AMR_AF=0.001879;gnomADe_ASJ_AF=3.827e-05;gnomADe_EAS_AF=0;gnomADe_FIN_AF=0;gnomADe_MID_AF=0.001214;gnomADe_NFE_AF=2.428e-05;gnomADe_REMAINING_AF=0.002053;gnomADe_SAS_AF=9.275e-05;gnomADg_AF=0.007867;gnomADg_AFR_AF=0.02713;gnomADg_AMI_AF=0;gnomADg_AMR_AF=0.003529;gnomADg_ASJ_AF=0;gnomADg_EAS_AF=0;gnomADg_FIN_AF=0;gnomADg_MID_AF=0;gnomADg_NFE_AF=0.0001029;gnomADg_REMAINING_AF=0.004748;gnomADg_SAS_AF=0;MAX_AF=0.0439;MAX_AF_POPS=AFR;CLIN_SIG=benign,likely_benign;PHENO=1</td>\n",
       "      <td>53307533</td>\n",
       "      <td>0.000793</td>\n",
       "      <td>0.007867</td>\n",
       "      <td>0.000793</td>\n",
       "      <td>ENSP00000209873.4:p.Gly533Arg</td>\n",
       "      <td>B</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.362112</td>\n",
       "      <td>20.9</td>\n",
       "      <td>2.951810</td>\n",
       "      <td>0.062144</td>\n",
       "      <td>0.420</td>\n",
       "      <td>0.45743</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.257</td>\n",
       "      <td>0.526710</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1192</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>0.977286</td>\n",
       "      <td>0.852</td>\n",
       "      <td>NaN</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20084</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_015665.6(AAAS):c.787T&gt;C (p.Ser263Pro)</td>\n",
       "      <td>8086</td>\n",
       "      <td>AAAS</td>\n",
       "      <td>HGNC:13666</td>\n",
       "      <td>Pathogenic/Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>121918550</td>\n",
       "      <td>-</td>\n",
       "      <td>RCV000005348|RCV000311283|RCV000415076|RCV000624696</td>\n",
       "      <td>MONDO:MONDO:0009279,MedGen:C0271742,OMIM:231550,Orphanet:869|MedGen:C3661900|Human Phenotype Ontology:HP:0002313,Human Phenotype Ontology:HP:0007191,MedGen:C0037771;Human Phenotype Ontology:HP:0001282,Human Phenotype Ontology:HP:0001347,Human Phenotype Ontology:HP:0006820,Human Phenotype Ontology:HP:0007184,Human Phenotype Ontology:HP:0007318,MONDO:MONDO:0007774,MedGen:C0151889,OMIM:145290;Human Phenotype Ontology:HP:0001352,Human Phenotype Ontology:HP:0003487,MedGen:C0034935|MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>Glucocorticoid deficiency with achalasia|not provided|Spastic paraparesis;Hyperreflexia;Babinski sign|Inborn genetic diseases</td>\n",
       "      <td>germline;unknown</td>\n",
       "      <td>germline</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>12</td>\n",
       "      <td>53309624</td>\n",
       "      <td>53309624</td>\n",
       "      <td>12q13.13</td>\n",
       "      <td>criteria provided, multiple submitters, no conflicts</td>\n",
       "      <td>11</td>\n",
       "      <td>ClinGen:CA117228,UniProtKB:Q9NRG9#VAR_012806,OMIM:605378.0007</td>\n",
       "      <td>3</td>\n",
       "      <td>5045</td>\n",
       "      <td>53309624</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Ser263Pro</td>\n",
       "      <td>S263P</td>\n",
       "      <td>2024</td>\n",
       "      <td>1.219186e+08</td>\n",
       "      <td>12:53309624</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000094914</td>\n",
       "      <td>ENST00000209873.9</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>932</td>\n",
       "      <td>787</td>\n",
       "      <td>263</td>\n",
       "      <td>S/P</td>\n",
       "      <td>Tca/Cca</td>\n",
       "      <td>rs121918550,CM010150</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;SYMBOL=AAAS;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:13666;CCDS=CCDS8856.1;ENSP=ENSP00000209873;SWISSPROT=Q9NRG9.206;UNIPARC=UPI0000039E40;UNIPROT_ISOFORM=Q9NRG9-1;HGVSc=ENST00000209873.9:c.787T&gt;C;HGVSp=ENSP00000209873.4:p.Ser263Pro;AF=0.0002;AFR_AF=0;AMR_AF=0;EAS_AF=0;EUR_AF=0.001;SAS_AF=0;gnomADe_AF=7.801e-05;gnomADe_AFR_AF=2.988e-05;gnomADe_AMR_AF=0;gnomADe_ASJ_AF=0;gnomADe_EAS_AF=0;gnomADe_FIN_AF=0.0005994;gnomADe_MID_AF=0;gnomADe_NFE_AF=6.925e-05;gnomADe_REMAINING_AF=6.627e-05;gnomADe_SAS_AF=0;gnomADg_AF=5.908e-05;gnomADg_AFR_AF=0;gnomADg_AMI_AF=0;gnomADg_AMR_AF=0;gnomADg_ASJ_AF=0;gnomADg_EAS_AF=0;gnomADg_FIN_AF=0.0002824;gnomADg_MID_AF=0;gnomADg_NFE_AF=8.819e-05;gnomADg_REMAINING_AF=0;gnomADg_SAS_AF=0;MAX_AF=0.001;MAX_AF_POPS=EUR;CLIN_SIG=pathogenic/likely_pathogenic,pathogenic;SOMATIC=0,1;PHENO=1,1</td>\n",
       "      <td>53309624</td>\n",
       "      <td>0.000078</td>\n",
       "      <td>0.000059</td>\n",
       "      <td>0.000078</td>\n",
       "      <td>ENSP00000209873.4:p.Ser263Pro</td>\n",
       "      <td>P</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1</td>\n",
       "      <td>0.992</td>\n",
       "      <td>1</td>\n",
       "      <td>0.258310</td>\n",
       "      <td>29.2</td>\n",
       "      <td>5.211912</td>\n",
       "      <td>0.980945</td>\n",
       "      <td>0.983</td>\n",
       "      <td>0.99270</td>\n",
       "      <td>1</td>\n",
       "      <td>0.951397</td>\n",
       "      <td>0.844</td>\n",
       "      <td>0.718980</td>\n",
       "      <td>1</td>\n",
       "      <td>0.9852</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>0.860283</td>\n",
       "      <td>0.458</td>\n",
       "      <td>NaN</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1312858</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_015665.6(AAAS):c.500C&gt;T (p.Ala167Val)</td>\n",
       "      <td>8086</td>\n",
       "      <td>AAAS</td>\n",
       "      <td>HGNC:13666</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>1017700992</td>\n",
       "      <td>-</td>\n",
       "      <td>RCV003557743|RCV004765890</td>\n",
       "      <td>MedGen:C3661900|MONDO:MONDO:0009279,MedGen:C0271742,OMIM:231550,Orphanet:869</td>\n",
       "      <td>not provided|Glucocorticoid deficiency with achalasia</td>\n",
       "      <td>germline</td>\n",
       "      <td>germline</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>12</td>\n",
       "      <td>53314796</td>\n",
       "      <td>53314796</td>\n",
       "      <td>12q13.13</td>\n",
       "      <td>criteria provided, multiple submitters, no conflicts</td>\n",
       "      <td>2</td>\n",
       "      <td>ClinGen:CA237335767</td>\n",
       "      <td>2</td>\n",
       "      <td>2735891</td>\n",
       "      <td>53314796</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Ala167Val</td>\n",
       "      <td>A167V</td>\n",
       "      <td>2024</td>\n",
       "      <td>1.017701e+09</td>\n",
       "      <td>12:53314796</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000094914</td>\n",
       "      <td>ENST00000209873.9</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>645</td>\n",
       "      <td>500</td>\n",
       "      <td>167</td>\n",
       "      <td>A/V</td>\n",
       "      <td>gCa/gTa</td>\n",
       "      <td>rs1017700992,CM065949</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;SYMBOL=AAAS;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:13666;CCDS=CCDS8856.1;ENSP=ENSP00000209873;SWISSPROT=Q9NRG9.206;UNIPARC=UPI0000039E40;UNIPROT_ISOFORM=Q9NRG9-1;HGVSc=ENST00000209873.9:c.500C&gt;T;HGVSp=ENSP00000209873.4:p.Ala167Val;gnomADe_AF=3.421e-06;gnomADe_AFR_AF=0;gnomADe_AMR_AF=0;gnomADe_ASJ_AF=0;gnomADe_EAS_AF=0;gnomADe_FIN_AF=0;gnomADe_MID_AF=0;gnomADe_NFE_AF=4.497e-06;gnomADe_REMAINING_AF=0;gnomADe_SAS_AF=0;MAX_AF=4.497e-06;MAX_AF_POPS=gnomADe_NFE;CLIN_SIG=pathogenic;SOMATIC=0,1;PHENO=1,1</td>\n",
       "      <td>53314796</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>ENSP00000209873.4:p.Ala167Val</td>\n",
       "      <td>P</td>\n",
       "      <td>1</td>\n",
       "      <td>0.03</td>\n",
       "      <td>1</td>\n",
       "      <td>0.969</td>\n",
       "      <td>1</td>\n",
       "      <td>0.571933</td>\n",
       "      <td>34.0</td>\n",
       "      <td>5.972388</td>\n",
       "      <td>0.998261</td>\n",
       "      <td>0.980</td>\n",
       "      <td>0.99026</td>\n",
       "      <td>1</td>\n",
       "      <td>0.853707</td>\n",
       "      <td>0.920</td>\n",
       "      <td>0.760144</td>\n",
       "      <td>1</td>\n",
       "      <td>0.8614</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>0.918544</td>\n",
       "      <td>0.766</td>\n",
       "      <td>NaN</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   #AlleleID                       Type                                       Name  GeneID GeneSymbol     HGNC_ID          ClinicalSignificance  ClinSigSimple  RS# (dbSNP) nsv/esv (dbVar)                                         RCVaccession                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   PhenotypeIDS                                                                                                                  PhenotypeList            Origin OriginSimple Assembly ChromosomeAccession Chromosome     Start      Stop Cytogenetic                                          ReviewStatus  NumberSubmitters                                                       OtherIDs  SubmitterCategories  VariationID  PositionVCF ReferenceAlleleVCF AlternateAlleleVCF Variant (3-letter) Variant  LastEvaluated (Year)  Uploaded_variation     Location Allele             Gene            Feature Feature_type       Consequence  cDNA_position  CDS_position  Protein_position Amino_acids   Codons     Existing_variation                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Extra  PositionVCF_dashed  gnomADe_AF  gnomADg_AF  gnomAD_AF                          HGVSp BinaryClinicalSignificance  SIFT_label  SIFT_score  PolyPhen_label  PolyPhen_score  BayesDel_label  BayesDel_score  CADD_PHRED_score  CADD_RAW_score  ClinPred_score  VEST4_score  VEST4_rankscore  EVE_label  EVE_score  REVEL_score  PrimateAI_score  AM_label  AM_score UniprotID  Envision_predictions  QAFI_predictions MetaRNN_score  MetaRNN_pred\n",
       "0     317931  single nucleotide variant  NM_015665.6(AAAS):c.1597G>A (p.Gly533Arg)    8086       AAAS  HGNC:13666          Benign/Likely benign              0     34451260               -                            RCV000343022|RCV000886859                                                                                                                                                                                                                                                                                                                                                                                                                                                   MONDO:MONDO:0009279,MedGen:C0271742,OMIM:231550,Orphanet:869|MedGen:C3661900                                                                          Glucocorticoid deficiency with achalasia|not provided          germline     germline   GRCh38        NC_000012.12         12  53307533  53307533    12q13.13  criteria provided, multiple submitters, no conflicts                 3                                              ClinGen:CA6598796                    2       309718     53307533                  C                  T          Gly533Arg   G533R                  2024        3.445126e+07  12:53307533      T  ENSG00000094914  ENST00000209873.9   Transcript  missense_variant           1742          1597               533         G/R  Ggg/Agg             rs34451260  IMPACT=MODERATE;STRAND=-1;SYMBOL=AAAS;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:13666;CCDS=CCDS8856.1;ENSP=ENSP00000209873;SWISSPROT=Q9NRG9.206;UNIPARC=UPI0000039E40;UNIPROT_ISOFORM=Q9NRG9-1;HGVSc=ENST00000209873.9:c.1597G>A;HGVSp=ENSP00000209873.4:p.Gly533Arg;AF=0.0124;AFR_AF=0.0439;AMR_AF=0.0043;EAS_AF=0;EUR_AF=0.001;SAS_AF=0;gnomADe_AF=0.0007928;gnomADe_AFR_AF=0.02712;gnomADe_AMR_AF=0.001879;gnomADe_ASJ_AF=3.827e-05;gnomADe_EAS_AF=0;gnomADe_FIN_AF=0;gnomADe_MID_AF=0.001214;gnomADe_NFE_AF=2.428e-05;gnomADe_REMAINING_AF=0.002053;gnomADe_SAS_AF=9.275e-05;gnomADg_AF=0.007867;gnomADg_AFR_AF=0.02713;gnomADg_AMI_AF=0;gnomADg_AMR_AF=0.003529;gnomADg_ASJ_AF=0;gnomADg_EAS_AF=0;gnomADg_FIN_AF=0;gnomADg_MID_AF=0;gnomADg_NFE_AF=0.0001029;gnomADg_REMAINING_AF=0.004748;gnomADg_SAS_AF=0;MAX_AF=0.0439;MAX_AF_POPS=AFR;CLIN_SIG=benign,likely_benign;PHENO=1            53307533    0.000793    0.007867   0.000793  ENSP00000209873.4:p.Gly533Arg                          B           1        0.00            <NA>           0.000               0       -0.362112              20.9        2.951810        0.062144        0.420          0.45743       <NA>        NaN        0.257         0.526710         0    0.1192    Q9NRG9              0.977286             0.852           NaN          <NA>\n",
       "1      20084  single nucleotide variant   NM_015665.6(AAAS):c.787T>C (p.Ser263Pro)    8086       AAAS  HGNC:13666  Pathogenic/Likely pathogenic              1    121918550               -  RCV000005348|RCV000311283|RCV000415076|RCV000624696  MONDO:MONDO:0009279,MedGen:C0271742,OMIM:231550,Orphanet:869|MedGen:C3661900|Human Phenotype Ontology:HP:0002313,Human Phenotype Ontology:HP:0007191,MedGen:C0037771;Human Phenotype Ontology:HP:0001282,Human Phenotype Ontology:HP:0001347,Human Phenotype Ontology:HP:0006820,Human Phenotype Ontology:HP:0007184,Human Phenotype Ontology:HP:0007318,MONDO:MONDO:0007774,MedGen:C0151889,OMIM:145290;Human Phenotype Ontology:HP:0001352,Human Phenotype Ontology:HP:0003487,MedGen:C0034935|MeSH:D030342,MedGen:C0950123  Glucocorticoid deficiency with achalasia|not provided|Spastic paraparesis;Hyperreflexia;Babinski sign|Inborn genetic diseases  germline;unknown     germline   GRCh38        NC_000012.12         12  53309624  53309624    12q13.13  criteria provided, multiple submitters, no conflicts                11  ClinGen:CA117228,UniProtKB:Q9NRG9#VAR_012806,OMIM:605378.0007                    3         5045     53309624                  A                  G          Ser263Pro   S263P                  2024        1.219186e+08  12:53309624      G  ENSG00000094914  ENST00000209873.9   Transcript  missense_variant            932           787               263         S/P  Tca/Cca   rs121918550,CM010150           IMPACT=MODERATE;STRAND=-1;SYMBOL=AAAS;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:13666;CCDS=CCDS8856.1;ENSP=ENSP00000209873;SWISSPROT=Q9NRG9.206;UNIPARC=UPI0000039E40;UNIPROT_ISOFORM=Q9NRG9-1;HGVSc=ENST00000209873.9:c.787T>C;HGVSp=ENSP00000209873.4:p.Ser263Pro;AF=0.0002;AFR_AF=0;AMR_AF=0;EAS_AF=0;EUR_AF=0.001;SAS_AF=0;gnomADe_AF=7.801e-05;gnomADe_AFR_AF=2.988e-05;gnomADe_AMR_AF=0;gnomADe_ASJ_AF=0;gnomADe_EAS_AF=0;gnomADe_FIN_AF=0.0005994;gnomADe_MID_AF=0;gnomADe_NFE_AF=6.925e-05;gnomADe_REMAINING_AF=6.627e-05;gnomADe_SAS_AF=0;gnomADg_AF=5.908e-05;gnomADg_AFR_AF=0;gnomADg_AMI_AF=0;gnomADg_AMR_AF=0;gnomADg_ASJ_AF=0;gnomADg_EAS_AF=0;gnomADg_FIN_AF=0.0002824;gnomADg_MID_AF=0;gnomADg_NFE_AF=8.819e-05;gnomADg_REMAINING_AF=0;gnomADg_SAS_AF=0;MAX_AF=0.001;MAX_AF_POPS=EUR;CLIN_SIG=pathogenic/likely_pathogenic,pathogenic;SOMATIC=0,1;PHENO=1,1            53309624    0.000078    0.000059   0.000078  ENSP00000209873.4:p.Ser263Pro                          P           1        0.00               1           0.992               1        0.258310              29.2        5.211912        0.980945        0.983          0.99270          1   0.951397        0.844         0.718980         1    0.9852    Q9NRG9              0.860283             0.458           NaN          <NA>\n",
       "2    1312858  single nucleotide variant   NM_015665.6(AAAS):c.500C>T (p.Ala167Val)    8086       AAAS  HGNC:13666                    Pathogenic              1   1017700992               -                            RCV003557743|RCV004765890                                                                                                                                                                                                                                                                                                                                                                                                                                                   MedGen:C3661900|MONDO:MONDO:0009279,MedGen:C0271742,OMIM:231550,Orphanet:869                                                                          not provided|Glucocorticoid deficiency with achalasia          germline     germline   GRCh38        NC_000012.12         12  53314796  53314796    12q13.13  criteria provided, multiple submitters, no conflicts                 2                                            ClinGen:CA237335767                    2      2735891     53314796                  G                  A          Ala167Val   A167V                  2024        1.017701e+09  12:53314796      A  ENSG00000094914  ENST00000209873.9   Transcript  missense_variant            645           500               167         A/V  gCa/gTa  rs1017700992,CM065949                                                                                                                                                                                                                                                                                                                                    IMPACT=MODERATE;STRAND=-1;SYMBOL=AAAS;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:13666;CCDS=CCDS8856.1;ENSP=ENSP00000209873;SWISSPROT=Q9NRG9.206;UNIPARC=UPI0000039E40;UNIPROT_ISOFORM=Q9NRG9-1;HGVSc=ENST00000209873.9:c.500C>T;HGVSp=ENSP00000209873.4:p.Ala167Val;gnomADe_AF=3.421e-06;gnomADe_AFR_AF=0;gnomADe_AMR_AF=0;gnomADe_ASJ_AF=0;gnomADe_EAS_AF=0;gnomADe_FIN_AF=0;gnomADe_MID_AF=0;gnomADe_NFE_AF=4.497e-06;gnomADe_REMAINING_AF=0;gnomADe_SAS_AF=0;MAX_AF=4.497e-06;MAX_AF_POPS=gnomADe_NFE;CLIN_SIG=pathogenic;SOMATIC=0,1;PHENO=1,1            53314796    0.000003         NaN   0.000003  ENSP00000209873.4:p.Ala167Val                          P           1        0.03               1           0.969               1        0.571933              34.0        5.972388        0.998261        0.980          0.99026          1   0.853707        0.920         0.760144         1    0.8614    Q9NRG9              0.918544             0.766           NaN          <NA>"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_allpreds.head(3)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aa4c9f97",
   "metadata": {},
   "source": [
    "### Humsavar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "cf72fe1f",
   "metadata": {},
   "outputs": [],
   "source": [
    "humsavar_allpreds = pd.read_csv('~/Desktop/analysis files/cleaned_Humsavar_ALL_preds.csv', sep=',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "ff40c57b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1108, 71)"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "humsavar_allpreds.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "dbad289c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Main gene name</th>\n",
       "      <th>Uniprot</th>\n",
       "      <th>FTId</th>\n",
       "      <th>AA change</th>\n",
       "      <th>Variant category</th>\n",
       "      <th>dbSNP</th>\n",
       "      <th>Disease name</th>\n",
       "      <th>Source</th>\n",
       "      <th>Variant</th>\n",
       "      <th>BinaryClinicalSignificance</th>\n",
       "      <th>MANE-Select ID</th>\n",
       "      <th>Ensembl ID</th>\n",
       "      <th>Protein ID</th>\n",
       "      <th>RefSeq Nucleotide ID</th>\n",
       "      <th>RefSeq Protein ID</th>\n",
       "      <th>Error</th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>REF_ALLELE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>TREMBL</th>\n",
       "      <th>UNIPARC</th>\n",
       "      <th>UNIPROT_ISOFORM</th>\n",
       "      <th>SIFT_label</th>\n",
       "      <th>SIFT_score</th>\n",
       "      <th>PolyPhen_label</th>\n",
       "      <th>PolyPhen_score</th>\n",
       "      <th>AF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "      <th>REVEL_score</th>\n",
       "      <th>BLOSUM62</th>\n",
       "      <th>EVE_label</th>\n",
       "      <th>EVE_score</th>\n",
       "      <th>CADD_PHRED_score</th>\n",
       "      <th>CADD_RAW_score</th>\n",
       "      <th>AM_label</th>\n",
       "      <th>AM_score</th>\n",
       "      <th>BayesDel_label</th>\n",
       "      <th>BayesDel_score</th>\n",
       "      <th>PrimateAI_label</th>\n",
       "      <th>PrimateAI_score</th>\n",
       "      <th>Uniprot_acc</th>\n",
       "      <th>Uniprot_entry</th>\n",
       "      <th>clinvar_id</th>\n",
       "      <th>clinvar_review</th>\n",
       "      <th>ClinPred_score</th>\n",
       "      <th>NaN_count</th>\n",
       "      <th>UniprotID</th>\n",
       "      <th>Envision_predictions</th>\n",
       "      <th>QAFI_predictions</th>\n",
       "      <th>MetaRNN_score</th>\n",
       "      <th>MetaRNN_pred</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABTB3</td>\n",
       "      <td>A6QL63</td>\n",
       "      <td>VAR_042534</td>\n",
       "      <td>p.Ala1002Asp</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs11610050</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2025</td>\n",
       "      <td>A1002D</td>\n",
       "      <td>B</td>\n",
       "      <td>ENST00000280758.10</td>\n",
       "      <td>ENST00000280758</td>\n",
       "      <td>ENSP00000280758.5</td>\n",
       "      <td>NM_001018072.2</td>\n",
       "      <td>NP_001018082.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs11610050</td>\n",
       "      <td>12:107651687-107651687</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>ABTB3</td>\n",
       "      <td>ENSG00000151136</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000280758.10</td>\n",
       "      <td>3512</td>\n",
       "      <td>3005</td>\n",
       "      <td>1002</td>\n",
       "      <td>A/D</td>\n",
       "      <td>gCt/gAt</td>\n",
       "      <td>rs11610050</td>\n",
       "      <td>C</td>\n",
       "      <td>HGNC:23844</td>\n",
       "      <td>ENSP00000280758</td>\n",
       "      <td>A6QL63.129</td>\n",
       "      <td>-</td>\n",
       "      <td>UPI000051901E</td>\n",
       "      <td>A6QL63-1</td>\n",
       "      <td>deleterious_low_confidence</td>\n",
       "      <td>0.00</td>\n",
       "      <td>probably_damaging</td>\n",
       "      <td>0.972</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.649</td>\n",
       "      <td>-2</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>31.0</td>\n",
       "      <td>5.339176</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>D</td>\n",
       "      <td>0.356338</td>\n",
       "      <td>D</td>\n",
       "      <td>0.848108232021</td>\n",
       "      <td>A6QL63,A6QL63-2,A6QL63-5,A6QL63-4</td>\n",
       "      <td>BTBDB_HUMAN,BTBDB_HUMAN,BTBDB_HUMAN,BTBDB_HUMAN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.999407649040222</td>\n",
       "      <td>0.0</td>\n",
       "      <td>A6QL63</td>\n",
       "      <td>0.772274</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABTB3</td>\n",
       "      <td>A6QL63</td>\n",
       "      <td>VAR_042535</td>\n",
       "      <td>p.Gly1076Ser</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs12303478</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2025</td>\n",
       "      <td>G1076S</td>\n",
       "      <td>B</td>\n",
       "      <td>ENST00000280758.10</td>\n",
       "      <td>ENST00000280758</td>\n",
       "      <td>ENSP00000280758.5</td>\n",
       "      <td>NM_001018072.2</td>\n",
       "      <td>NP_001018082.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs12303478</td>\n",
       "      <td>12:107657629-107657629</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>ABTB3</td>\n",
       "      <td>ENSG00000151136</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000280758.10</td>\n",
       "      <td>3733</td>\n",
       "      <td>3226</td>\n",
       "      <td>1076</td>\n",
       "      <td>G/S</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>rs12303478,COSV104383452</td>\n",
       "      <td>G</td>\n",
       "      <td>HGNC:23844</td>\n",
       "      <td>ENSP00000280758</td>\n",
       "      <td>A6QL63.129</td>\n",
       "      <td>-</td>\n",
       "      <td>UPI000051901E</td>\n",
       "      <td>A6QL63-1</td>\n",
       "      <td>tolerated_low_confidence</td>\n",
       "      <td>0.09</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.049</td>\n",
       "      <td>0.1048</td>\n",
       "      <td>-</td>\n",
       "      <td>0,1</td>\n",
       "      <td>0,1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.072</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>22.4</td>\n",
       "      <td>3.396815</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.0921</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.583583</td>\n",
       "      <td>T</td>\n",
       "      <td>0.613067984581</td>\n",
       "      <td>A6QL63,A6QL63-2,A6QL63-5,A6QL63-4</td>\n",
       "      <td>BTBDB_HUMAN,BTBDB_HUMAN,BTBDB_HUMAN,BTBDB_HUMAN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.0157182390193226</td>\n",
       "      <td>0.0</td>\n",
       "      <td>A6QL63</td>\n",
       "      <td>0.978557</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABTB3</td>\n",
       "      <td>A6QL63</td>\n",
       "      <td>VAR_055560</td>\n",
       "      <td>p.Gly448Ser</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs1558781</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2025</td>\n",
       "      <td>G448S</td>\n",
       "      <td>B</td>\n",
       "      <td>ENST00000280758.10</td>\n",
       "      <td>ENST00000280758</td>\n",
       "      <td>ENSP00000280758.5</td>\n",
       "      <td>NM_001018072.2</td>\n",
       "      <td>NP_001018082.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1558781</td>\n",
       "      <td>12:107543991-107543991</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>ABTB3</td>\n",
       "      <td>ENSG00000151136</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000280758.10</td>\n",
       "      <td>1849</td>\n",
       "      <td>1342</td>\n",
       "      <td>448</td>\n",
       "      <td>G/S</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>rs1558781,COSV99774826</td>\n",
       "      <td>G</td>\n",
       "      <td>HGNC:23844</td>\n",
       "      <td>ENSP00000280758</td>\n",
       "      <td>A6QL63.129</td>\n",
       "      <td>-</td>\n",
       "      <td>UPI000051901E</td>\n",
       "      <td>A6QL63-1</td>\n",
       "      <td>tolerated_low_confidence</td>\n",
       "      <td>0.43</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.022</td>\n",
       "      <td>0.1763</td>\n",
       "      <td>-</td>\n",
       "      <td>0,1</td>\n",
       "      <td>0,1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.102</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>22.3</td>\n",
       "      <td>3.373019</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.0774</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.626984</td>\n",
       "      <td>T</td>\n",
       "      <td>0.485189527273</td>\n",
       "      <td>A6QL63,A6QL63-2,A6QL63-3,H0YHR1,C9JV03</td>\n",
       "      <td>BTBDB_HUMAN,BTBDB_HUMAN,BTBDB_HUMAN,H0YHR1_HUMAN,C9JV03_HUMAN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.0154826749531858</td>\n",
       "      <td>0.0</td>\n",
       "      <td>A6QL63</td>\n",
       "      <td>0.983750</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Main gene name Uniprot        FTId     AA change Variant category       dbSNP Disease name  Source Variant BinaryClinicalSignificance      MANE-Select ID       Ensembl ID         Protein ID RefSeq Nucleotide ID RefSeq Protein ID  Error #Uploaded_variation                Location Allele       Consequence    IMPACT GeneSymbol             Gene Feature_type             Feature  cDNA_position  CDS_position  Protein_position Amino_acids   Codons        Existing_variation REF_ALLELE     HGNC_ID             ENSP   SWISSPROT TREMBL        UNIPARC UNIPROT_ISOFORM                  SIFT_label  SIFT_score     PolyPhen_label  PolyPhen_score      AF CLIN_SIG SOMATIC PHENO PUBMED TRANSCRIPTION_FACTORS REVEL_score  BLOSUM62 EVE_label EVE_score  CADD_PHRED_score  CADD_RAW_score           AM_label AM_score BayesDel_label BayesDel_score PrimateAI_label PrimateAI_score                             Uniprot_acc                                                  Uniprot_entry clinvar_id clinvar_review      ClinPred_score  NaN_count UniprotID  Envision_predictions  QAFI_predictions MetaRNN_score MetaRNN_pred\n",
       "0          ABTB3  A6QL63  VAR_042534  p.Ala1002Asp             LB/B  rs11610050          NaN    2025  A1002D                          B  ENST00000280758.10  ENST00000280758  ENSP00000280758.5       NM_001018072.2    NP_001018082.1    NaN          rs11610050  12:107651687-107651687      A  missense_variant  MODERATE      ABTB3  ENSG00000151136   Transcript  ENST00000280758.10           3512          3005              1002         A/D  gCt/gAt                rs11610050          C  HGNC:23844  ENSP00000280758  A6QL63.129      -  UPI000051901E        A6QL63-1  deleterious_low_confidence        0.00  probably_damaging           0.972       -        -       -     -      -                     -       0.649        -2         -         -              31.0        5.339176  likely_pathogenic   0.9994              D       0.356338               D  0.848108232021       A6QL63,A6QL63-2,A6QL63-5,A6QL63-4                BTBDB_HUMAN,BTBDB_HUMAN,BTBDB_HUMAN,BTBDB_HUMAN          -              -   0.999407649040222        0.0    A6QL63              0.772274               NaN           NaN          NaN\n",
       "1          ABTB3  A6QL63  VAR_042535  p.Gly1076Ser             LB/B  rs12303478          NaN    2025  G1076S                          B  ENST00000280758.10  ENST00000280758  ENSP00000280758.5       NM_001018072.2    NP_001018082.1    NaN          rs12303478  12:107657629-107657629      A  missense_variant  MODERATE      ABTB3  ENSG00000151136   Transcript  ENST00000280758.10           3733          3226              1076         G/S  Ggc/Agc  rs12303478,COSV104383452          G  HGNC:23844  ENSP00000280758  A6QL63.129      -  UPI000051901E        A6QL63-1    tolerated_low_confidence        0.09             benign           0.049  0.1048        -     0,1   0,1      -                     -       0.072         0         -         -              22.4        3.396815      likely_benign   0.0921              T      -0.583583               T  0.613067984581       A6QL63,A6QL63-2,A6QL63-5,A6QL63-4                BTBDB_HUMAN,BTBDB_HUMAN,BTBDB_HUMAN,BTBDB_HUMAN          -              -  0.0157182390193226        0.0    A6QL63              0.978557               NaN           NaN          NaN\n",
       "2          ABTB3  A6QL63  VAR_055560   p.Gly448Ser             LB/B   rs1558781          NaN    2025   G448S                          B  ENST00000280758.10  ENST00000280758  ENSP00000280758.5       NM_001018072.2    NP_001018082.1    NaN           rs1558781  12:107543991-107543991      A  missense_variant  MODERATE      ABTB3  ENSG00000151136   Transcript  ENST00000280758.10           1849          1342               448         G/S  Ggc/Agc    rs1558781,COSV99774826          G  HGNC:23844  ENSP00000280758  A6QL63.129      -  UPI000051901E        A6QL63-1    tolerated_low_confidence        0.43             benign           0.022  0.1763        -     0,1   0,1      -                     -       0.102         0         -         -              22.3        3.373019      likely_benign   0.0774              T      -0.626984               T  0.485189527273  A6QL63,A6QL63-2,A6QL63-3,H0YHR1,C9JV03  BTBDB_HUMAN,BTBDB_HUMAN,BTBDB_HUMAN,H0YHR1_HUMAN,C9JV03_HUMAN          -              -  0.0154826749531858        0.0    A6QL63              0.983750               NaN           NaN          NaN"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "humsavar_allpreds.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "f5bc6e04",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "BayesDel_label            0\n",
       "BayesDel_score            0\n",
       "CADD_PHRED_score          0\n",
       "CADD_RAW_score            0\n",
       "EVE_label                 0\n",
       "EVE_score                 0\n",
       "REVEL_score               0\n",
       "AM_label                  0\n",
       "AM_score                  0\n",
       "Envision_predictions    246\n",
       "QAFI_predictions        834\n",
       "MetaRNN_score           716\n",
       "MetaRNN_pred            716\n",
       "dtype: int64"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "humsavar_allpreds[['BayesDel_label', 'BayesDel_score', 'CADD_PHRED_score', 'CADD_RAW_score', 'EVE_label', 'EVE_score', 'REVEL_score', 'AM_label', 'AM_score', 'Envision_predictions', 'QAFI_predictions', 'MetaRNN_score', 'MetaRNN_pred']].isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "id": "5abf5077",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SIFT_label: ['deleterious_low_confidence' 'tolerated_low_confidence' 'deleterious'\n",
      " 'tolerated' '-']\n",
      "PolyPhen_label: ['probably_damaging' 'benign' 'possibly_damaging' 'unknown' '-']\n",
      "EVE_label: ['-' 'Pathogenic' 'Benign' 'Uncertain']\n",
      "AM_label: ['likely_pathogenic' 'likely_benign' 'ambiguous' '-']\n",
      "BayesDel_label: ['D' 'T' '-']\n",
      "PrimateAI_label: ['D' 'T' '-']\n",
      "MetaRNN_pred: [nan 'D' 'T' '.']\n"
     ]
    }
   ],
   "source": [
    "for col in humsavar_allpreds.columns:\n",
    "    if col.endswith('label') or col.endswith('pred'):\n",
    "        print(f\"{col}: {humsavar_allpreds[col].unique()}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "id": "085f1a4b",
   "metadata": {},
   "outputs": [],
   "source": [
    "mappings = {\n",
    "    'SIFT_label': {'deleterious': 1, 'deleterious_low_confidence': 1, 'tolerated': 0, 'tolerated_low_confidence': 0},\n",
    "    'PolyPhen_label': {'probably_damaging': 1, 'possibly_damaging': 1, 'benign': 0},\n",
    "    'BayesDel_label': {'T': 0, 'D': 1},\n",
    "    'EVE_label': {'Benign': 0, 'Pathogenic': 1},\n",
    "    'AM_label': {'likely_benign': 0, 'likely_pathogenic': 1},\n",
    "    'MetaRNN_pred': {'T': 0, 'D': 1}\n",
    "}\n",
    "\n",
    "for col, mapping in mappings.items():\n",
    "    humsavar_allpreds[col] = humsavar_allpreds[col].map(mapping).astype('Int64')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "id": "fbcee41c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Main gene name</th>\n",
       "      <th>Uniprot</th>\n",
       "      <th>FTId</th>\n",
       "      <th>AA change</th>\n",
       "      <th>Variant category</th>\n",
       "      <th>dbSNP</th>\n",
       "      <th>Disease name</th>\n",
       "      <th>Source</th>\n",
       "      <th>Variant</th>\n",
       "      <th>BinaryClinicalSignificance</th>\n",
       "      <th>MANE-Select ID</th>\n",
       "      <th>Ensembl ID</th>\n",
       "      <th>Protein ID</th>\n",
       "      <th>RefSeq Nucleotide ID</th>\n",
       "      <th>RefSeq Protein ID</th>\n",
       "      <th>Error</th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>REF_ALLELE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>TREMBL</th>\n",
       "      <th>UNIPARC</th>\n",
       "      <th>UNIPROT_ISOFORM</th>\n",
       "      <th>SIFT_label</th>\n",
       "      <th>SIFT_score</th>\n",
       "      <th>PolyPhen_label</th>\n",
       "      <th>PolyPhen_score</th>\n",
       "      <th>AF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "      <th>REVEL_score</th>\n",
       "      <th>BLOSUM62</th>\n",
       "      <th>EVE_label</th>\n",
       "      <th>EVE_score</th>\n",
       "      <th>CADD_PHRED_score</th>\n",
       "      <th>CADD_RAW_score</th>\n",
       "      <th>AM_label</th>\n",
       "      <th>AM_score</th>\n",
       "      <th>BayesDel_label</th>\n",
       "      <th>BayesDel_score</th>\n",
       "      <th>PrimateAI_label</th>\n",
       "      <th>PrimateAI_score</th>\n",
       "      <th>Uniprot_acc</th>\n",
       "      <th>Uniprot_entry</th>\n",
       "      <th>clinvar_id</th>\n",
       "      <th>clinvar_review</th>\n",
       "      <th>ClinPred_score</th>\n",
       "      <th>NaN_count</th>\n",
       "      <th>UniprotID</th>\n",
       "      <th>Envision_predictions</th>\n",
       "      <th>QAFI_predictions</th>\n",
       "      <th>MetaRNN_score</th>\n",
       "      <th>MetaRNN_pred</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABTB3</td>\n",
       "      <td>A6QL63</td>\n",
       "      <td>VAR_042534</td>\n",
       "      <td>p.Ala1002Asp</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs11610050</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2025</td>\n",
       "      <td>A1002D</td>\n",
       "      <td>B</td>\n",
       "      <td>ENST00000280758.10</td>\n",
       "      <td>ENST00000280758</td>\n",
       "      <td>ENSP00000280758.5</td>\n",
       "      <td>NM_001018072.2</td>\n",
       "      <td>NP_001018082.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs11610050</td>\n",
       "      <td>12:107651687-107651687</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>ABTB3</td>\n",
       "      <td>ENSG00000151136</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000280758.10</td>\n",
       "      <td>3512</td>\n",
       "      <td>3005</td>\n",
       "      <td>1002</td>\n",
       "      <td>A/D</td>\n",
       "      <td>gCt/gAt</td>\n",
       "      <td>rs11610050</td>\n",
       "      <td>C</td>\n",
       "      <td>HGNC:23844</td>\n",
       "      <td>ENSP00000280758</td>\n",
       "      <td>A6QL63.129</td>\n",
       "      <td>-</td>\n",
       "      <td>UPI000051901E</td>\n",
       "      <td>A6QL63-1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1</td>\n",
       "      <td>0.972</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.649</td>\n",
       "      <td>-2</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>-</td>\n",
       "      <td>31.0</td>\n",
       "      <td>5.339176</td>\n",
       "      <td>1</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>1</td>\n",
       "      <td>0.356338</td>\n",
       "      <td>D</td>\n",
       "      <td>0.848108232021</td>\n",
       "      <td>A6QL63,A6QL63-2,A6QL63-5,A6QL63-4</td>\n",
       "      <td>BTBDB_HUMAN,BTBDB_HUMAN,BTBDB_HUMAN,BTBDB_HUMAN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.999407649040222</td>\n",
       "      <td>0.0</td>\n",
       "      <td>A6QL63</td>\n",
       "      <td>0.772274</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABTB3</td>\n",
       "      <td>A6QL63</td>\n",
       "      <td>VAR_042535</td>\n",
       "      <td>p.Gly1076Ser</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs12303478</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2025</td>\n",
       "      <td>G1076S</td>\n",
       "      <td>B</td>\n",
       "      <td>ENST00000280758.10</td>\n",
       "      <td>ENST00000280758</td>\n",
       "      <td>ENSP00000280758.5</td>\n",
       "      <td>NM_001018072.2</td>\n",
       "      <td>NP_001018082.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs12303478</td>\n",
       "      <td>12:107657629-107657629</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>ABTB3</td>\n",
       "      <td>ENSG00000151136</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000280758.10</td>\n",
       "      <td>3733</td>\n",
       "      <td>3226</td>\n",
       "      <td>1076</td>\n",
       "      <td>G/S</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>rs12303478,COSV104383452</td>\n",
       "      <td>G</td>\n",
       "      <td>HGNC:23844</td>\n",
       "      <td>ENSP00000280758</td>\n",
       "      <td>A6QL63.129</td>\n",
       "      <td>-</td>\n",
       "      <td>UPI000051901E</td>\n",
       "      <td>A6QL63-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0</td>\n",
       "      <td>0.049</td>\n",
       "      <td>0.1048</td>\n",
       "      <td>-</td>\n",
       "      <td>0,1</td>\n",
       "      <td>0,1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.072</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>-</td>\n",
       "      <td>22.4</td>\n",
       "      <td>3.396815</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0921</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.583583</td>\n",
       "      <td>T</td>\n",
       "      <td>0.613067984581</td>\n",
       "      <td>A6QL63,A6QL63-2,A6QL63-5,A6QL63-4</td>\n",
       "      <td>BTBDB_HUMAN,BTBDB_HUMAN,BTBDB_HUMAN,BTBDB_HUMAN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.0157182390193226</td>\n",
       "      <td>0.0</td>\n",
       "      <td>A6QL63</td>\n",
       "      <td>0.978557</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABTB3</td>\n",
       "      <td>A6QL63</td>\n",
       "      <td>VAR_055560</td>\n",
       "      <td>p.Gly448Ser</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs1558781</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2025</td>\n",
       "      <td>G448S</td>\n",
       "      <td>B</td>\n",
       "      <td>ENST00000280758.10</td>\n",
       "      <td>ENST00000280758</td>\n",
       "      <td>ENSP00000280758.5</td>\n",
       "      <td>NM_001018072.2</td>\n",
       "      <td>NP_001018082.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1558781</td>\n",
       "      <td>12:107543991-107543991</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>ABTB3</td>\n",
       "      <td>ENSG00000151136</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000280758.10</td>\n",
       "      <td>1849</td>\n",
       "      <td>1342</td>\n",
       "      <td>448</td>\n",
       "      <td>G/S</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>rs1558781,COSV99774826</td>\n",
       "      <td>G</td>\n",
       "      <td>HGNC:23844</td>\n",
       "      <td>ENSP00000280758</td>\n",
       "      <td>A6QL63.129</td>\n",
       "      <td>-</td>\n",
       "      <td>UPI000051901E</td>\n",
       "      <td>A6QL63-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0</td>\n",
       "      <td>0.022</td>\n",
       "      <td>0.1763</td>\n",
       "      <td>-</td>\n",
       "      <td>0,1</td>\n",
       "      <td>0,1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.102</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>-</td>\n",
       "      <td>22.3</td>\n",
       "      <td>3.373019</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0774</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.626984</td>\n",
       "      <td>T</td>\n",
       "      <td>0.485189527273</td>\n",
       "      <td>A6QL63,A6QL63-2,A6QL63-3,H0YHR1,C9JV03</td>\n",
       "      <td>BTBDB_HUMAN,BTBDB_HUMAN,BTBDB_HUMAN,H0YHR1_HUMAN,C9JV03_HUMAN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.0154826749531858</td>\n",
       "      <td>0.0</td>\n",
       "      <td>A6QL63</td>\n",
       "      <td>0.983750</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Main gene name Uniprot        FTId     AA change Variant category       dbSNP Disease name  Source Variant BinaryClinicalSignificance      MANE-Select ID       Ensembl ID         Protein ID RefSeq Nucleotide ID RefSeq Protein ID  Error #Uploaded_variation                Location Allele       Consequence    IMPACT GeneSymbol             Gene Feature_type             Feature  cDNA_position  CDS_position  Protein_position Amino_acids   Codons        Existing_variation REF_ALLELE     HGNC_ID             ENSP   SWISSPROT TREMBL        UNIPARC UNIPROT_ISOFORM  SIFT_label  SIFT_score  PolyPhen_label  PolyPhen_score      AF CLIN_SIG SOMATIC PHENO PUBMED TRANSCRIPTION_FACTORS REVEL_score  BLOSUM62  EVE_label EVE_score  CADD_PHRED_score  CADD_RAW_score  AM_label AM_score  BayesDel_label BayesDel_score PrimateAI_label PrimateAI_score                             Uniprot_acc                                                  Uniprot_entry clinvar_id clinvar_review      ClinPred_score  NaN_count UniprotID  Envision_predictions  QAFI_predictions MetaRNN_score  MetaRNN_pred\n",
       "0          ABTB3  A6QL63  VAR_042534  p.Ala1002Asp             LB/B  rs11610050          NaN    2025  A1002D                          B  ENST00000280758.10  ENST00000280758  ENSP00000280758.5       NM_001018072.2    NP_001018082.1    NaN          rs11610050  12:107651687-107651687      A  missense_variant  MODERATE      ABTB3  ENSG00000151136   Transcript  ENST00000280758.10           3512          3005              1002         A/D  gCt/gAt                rs11610050          C  HGNC:23844  ENSP00000280758  A6QL63.129      -  UPI000051901E        A6QL63-1           1        0.00               1           0.972       -        -       -     -      -                     -       0.649        -2       <NA>         -              31.0        5.339176         1   0.9994               1       0.356338               D  0.848108232021       A6QL63,A6QL63-2,A6QL63-5,A6QL63-4                BTBDB_HUMAN,BTBDB_HUMAN,BTBDB_HUMAN,BTBDB_HUMAN          -              -   0.999407649040222        0.0    A6QL63              0.772274               NaN           NaN          <NA>\n",
       "1          ABTB3  A6QL63  VAR_042535  p.Gly1076Ser             LB/B  rs12303478          NaN    2025  G1076S                          B  ENST00000280758.10  ENST00000280758  ENSP00000280758.5       NM_001018072.2    NP_001018082.1    NaN          rs12303478  12:107657629-107657629      A  missense_variant  MODERATE      ABTB3  ENSG00000151136   Transcript  ENST00000280758.10           3733          3226              1076         G/S  Ggc/Agc  rs12303478,COSV104383452          G  HGNC:23844  ENSP00000280758  A6QL63.129      -  UPI000051901E        A6QL63-1           0        0.09               0           0.049  0.1048        -     0,1   0,1      -                     -       0.072         0       <NA>         -              22.4        3.396815         0   0.0921               0      -0.583583               T  0.613067984581       A6QL63,A6QL63-2,A6QL63-5,A6QL63-4                BTBDB_HUMAN,BTBDB_HUMAN,BTBDB_HUMAN,BTBDB_HUMAN          -              -  0.0157182390193226        0.0    A6QL63              0.978557               NaN           NaN          <NA>\n",
       "2          ABTB3  A6QL63  VAR_055560   p.Gly448Ser             LB/B   rs1558781          NaN    2025   G448S                          B  ENST00000280758.10  ENST00000280758  ENSP00000280758.5       NM_001018072.2    NP_001018082.1    NaN           rs1558781  12:107543991-107543991      A  missense_variant  MODERATE      ABTB3  ENSG00000151136   Transcript  ENST00000280758.10           1849          1342               448         G/S  Ggc/Agc    rs1558781,COSV99774826          G  HGNC:23844  ENSP00000280758  A6QL63.129      -  UPI000051901E        A6QL63-1           0        0.43               0           0.022  0.1763        -     0,1   0,1      -                     -       0.102         0       <NA>         -              22.3        3.373019         0   0.0774               0      -0.626984               T  0.485189527273  A6QL63,A6QL63-2,A6QL63-3,H0YHR1,C9JV03  BTBDB_HUMAN,BTBDB_HUMAN,BTBDB_HUMAN,H0YHR1_HUMAN,C9JV03_HUMAN          -              -  0.0154826749531858        0.0    A6QL63              0.983750               NaN           NaN          <NA>"
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "humsavar_allpreds.head(3)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "87e9bebe",
   "metadata": {},
   "source": [
    "### Simplify ClinHum data to single tidy dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "id": "fee832a0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['LB/B', 'LP/P'], dtype=object)"
      ]
     },
     "execution_count": 139,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "humsavar_allpreds[\"Variant category\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "id": "73a6f05e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# columns in ClinVar order\n",
    "cols_to_keep = [\n",
    "    'GeneSymbol', 'ClinSigSimple', 'Variant', 'UniprotID',\n",
    "    'SIFT_label', 'SIFT_score', 'PolyPhen_label', 'PolyPhen_score',\n",
    "    'BayesDel_label', 'BayesDel_score', 'CADD_PHRED_score', 'CADD_RAW_score',\n",
    "    'ClinPred_score', 'VEST4_score', 'EVE_label', 'EVE_score',\n",
    "    'REVEL_score', 'PrimateAI_score', 'AM_label', 'AM_score',\n",
    "    'Envision_predictions', 'QAFI_predictions', 'MetaRNN_score', 'MetaRNN_pred'\n",
    "]\n",
    "\n",
    "clinvar_subset = clinvar_allpreds[cols_to_keep]\n",
    "\n",
    "test = humsavar_allpreds.rename(columns={'Variant category': 'ClinSigSimple',})\n",
    "test['ClinSigSimple'] = test['ClinSigSimple'].map({'LB/B': 0, 'LP/P': 1})\n",
    "test['VEST4_score'] = np.nan\n",
    "\n",
    "humsavar_subset = test[cols_to_keep]\n",
    "\n",
    "combined = pd.concat([clinvar_subset, humsavar_subset], ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "1ae6710d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>Variant</th>\n",
       "      <th>UniprotID</th>\n",
       "      <th>SIFT_label</th>\n",
       "      <th>SIFT_score</th>\n",
       "      <th>PolyPhen_label</th>\n",
       "      <th>PolyPhen_score</th>\n",
       "      <th>BayesDel_label</th>\n",
       "      <th>BayesDel_score</th>\n",
       "      <th>CADD_PHRED_score</th>\n",
       "      <th>CADD_RAW_score</th>\n",
       "      <th>ClinPred_score</th>\n",
       "      <th>VEST4_score</th>\n",
       "      <th>EVE_label</th>\n",
       "      <th>EVE_score</th>\n",
       "      <th>REVEL_score</th>\n",
       "      <th>PrimateAI_score</th>\n",
       "      <th>AM_label</th>\n",
       "      <th>AM_score</th>\n",
       "      <th>Envision_predictions</th>\n",
       "      <th>QAFI_predictions</th>\n",
       "      <th>MetaRNN_score</th>\n",
       "      <th>MetaRNN_pred</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>0</td>\n",
       "      <td>G533R</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.362112</td>\n",
       "      <td>20.9</td>\n",
       "      <td>2.951810</td>\n",
       "      <td>0.062144</td>\n",
       "      <td>0.420</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.257</td>\n",
       "      <td>0.52671</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1192</td>\n",
       "      <td>0.977286</td>\n",
       "      <td>0.852</td>\n",
       "      <td>NaN</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>1</td>\n",
       "      <td>S263P</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1</td>\n",
       "      <td>0.992</td>\n",
       "      <td>1</td>\n",
       "      <td>0.25831</td>\n",
       "      <td>29.2</td>\n",
       "      <td>5.211912</td>\n",
       "      <td>0.980945</td>\n",
       "      <td>0.983</td>\n",
       "      <td>1</td>\n",
       "      <td>0.951397</td>\n",
       "      <td>0.844</td>\n",
       "      <td>0.71898</td>\n",
       "      <td>1</td>\n",
       "      <td>0.9852</td>\n",
       "      <td>0.860283</td>\n",
       "      <td>0.458</td>\n",
       "      <td>NaN</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>1</td>\n",
       "      <td>A167V</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>1</td>\n",
       "      <td>0.03</td>\n",
       "      <td>1</td>\n",
       "      <td>0.969</td>\n",
       "      <td>1</td>\n",
       "      <td>0.571933</td>\n",
       "      <td>34.0</td>\n",
       "      <td>5.972388</td>\n",
       "      <td>0.998261</td>\n",
       "      <td>0.980</td>\n",
       "      <td>1</td>\n",
       "      <td>0.853707</td>\n",
       "      <td>0.92</td>\n",
       "      <td>0.760144</td>\n",
       "      <td>1</td>\n",
       "      <td>0.8614</td>\n",
       "      <td>0.918544</td>\n",
       "      <td>0.766</td>\n",
       "      <td>NaN</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>1</td>\n",
       "      <td>H160R</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1</td>\n",
       "      <td>0.952</td>\n",
       "      <td>1</td>\n",
       "      <td>0.553083</td>\n",
       "      <td>27.5</td>\n",
       "      <td>4.924293</td>\n",
       "      <td>0.99862</td>\n",
       "      <td>0.985</td>\n",
       "      <td>1</td>\n",
       "      <td>0.963709</td>\n",
       "      <td>0.919</td>\n",
       "      <td>0.675616</td>\n",
       "      <td>1</td>\n",
       "      <td>0.9973</td>\n",
       "      <td>0.828964</td>\n",
       "      <td>0.536</td>\n",
       "      <td>NaN</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>1</td>\n",
       "      <td>R155H</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>1</td>\n",
       "      <td>0.02</td>\n",
       "      <td>1</td>\n",
       "      <td>0.980</td>\n",
       "      <td>1</td>\n",
       "      <td>0.449379</td>\n",
       "      <td>29.4</td>\n",
       "      <td>5.229742</td>\n",
       "      <td>0.99574</td>\n",
       "      <td>0.898</td>\n",
       "      <td>1</td>\n",
       "      <td>0.717298</td>\n",
       "      <td>0.909</td>\n",
       "      <td>0.770769</td>\n",
       "      <td>1</td>\n",
       "      <td>0.6977</td>\n",
       "      <td>0.897079</td>\n",
       "      <td>0.647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  GeneSymbol  ClinSigSimple Variant UniprotID  SIFT_label  SIFT_score  PolyPhen_label  PolyPhen_score  BayesDel_label BayesDel_score  CADD_PHRED_score  CADD_RAW_score ClinPred_score  VEST4_score  EVE_label EVE_score REVEL_score PrimateAI_score  AM_label AM_score  Envision_predictions  QAFI_predictions MetaRNN_score  MetaRNN_pred\n",
       "0       AAAS              0   G533R    Q9NRG9           1        0.00            <NA>           0.000               0      -0.362112              20.9        2.951810       0.062144        0.420       <NA>       NaN       0.257         0.52671         0   0.1192              0.977286             0.852           NaN          <NA>\n",
       "1       AAAS              1   S263P    Q9NRG9           1        0.00               1           0.992               1        0.25831              29.2        5.211912       0.980945        0.983          1  0.951397       0.844         0.71898         1   0.9852              0.860283             0.458           NaN          <NA>\n",
       "2       AAAS              1   A167V    Q9NRG9           1        0.03               1           0.969               1       0.571933              34.0        5.972388       0.998261        0.980          1  0.853707        0.92        0.760144         1   0.8614              0.918544             0.766           NaN          <NA>\n",
       "3       AAAS              1   H160R    Q9NRG9           1        0.00               1           0.952               1       0.553083              27.5        4.924293        0.99862        0.985          1  0.963709       0.919        0.675616         1   0.9973              0.828964             0.536           NaN          <NA>\n",
       "4       AAAS              1   R155H    Q9NRG9           1        0.02               1           0.980               1       0.449379              29.4        5.229742        0.99574        0.898          1  0.717298       0.909        0.770769         1   0.6977              0.897079             0.647           NaN          <NA>"
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "5b596488",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GeneSymbol                  0\n",
       "ClinSigSimple               0\n",
       "Variant                     0\n",
       "UniprotID                   0\n",
       "SIFT_label               1099\n",
       "SIFT_score               1099\n",
       "PolyPhen_label           6883\n",
       "PolyPhen_score           3322\n",
       "BayesDel_label             69\n",
       "BayesDel_score             55\n",
       "CADD_PHRED_score            0\n",
       "CADD_RAW_score              0\n",
       "ClinPred_score            141\n",
       "VEST4_score              1108\n",
       "EVE_label               22852\n",
       "EVE_score               15760\n",
       "REVEL_score              3876\n",
       "PrimateAI_score           510\n",
       "AM_label                 6953\n",
       "AM_score                 2879\n",
       "Envision_predictions     3583\n",
       "QAFI_predictions        17261\n",
       "MetaRNN_score           37279\n",
       "MetaRNN_pred            37388\n",
       "dtype: int64"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined.isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "id": "193172df",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(47158, 24)"
      ]
     },
     "execution_count": 143,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "id": "642a0872",
   "metadata": {},
   "outputs": [],
   "source": [
    "missing_values = ['<NA>', 'NA', 'NaN', 'nan', 'None', '', 'null']\n",
    "for col in combined.columns:\n",
    "    combined[col] = combined[col].astype(str).replace(missing_values, np.nan)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "d658eed9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>Variant</th>\n",
       "      <th>UniprotID</th>\n",
       "      <th>SIFT_label</th>\n",
       "      <th>SIFT_score</th>\n",
       "      <th>PolyPhen_label</th>\n",
       "      <th>PolyPhen_score</th>\n",
       "      <th>BayesDel_label</th>\n",
       "      <th>BayesDel_score</th>\n",
       "      <th>CADD_PHRED_score</th>\n",
       "      <th>CADD_RAW_score</th>\n",
       "      <th>ClinPred_score</th>\n",
       "      <th>VEST4_score</th>\n",
       "      <th>EVE_label</th>\n",
       "      <th>EVE_score</th>\n",
       "      <th>REVEL_score</th>\n",
       "      <th>PrimateAI_score</th>\n",
       "      <th>AM_label</th>\n",
       "      <th>AM_score</th>\n",
       "      <th>Envision_predictions</th>\n",
       "      <th>QAFI_predictions</th>\n",
       "      <th>MetaRNN_score</th>\n",
       "      <th>MetaRNN_pred</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>0</td>\n",
       "      <td>G533R</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.362112</td>\n",
       "      <td>20.9</td>\n",
       "      <td>2.951810</td>\n",
       "      <td>0.062144</td>\n",
       "      <td>0.420</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.257</td>\n",
       "      <td>0.52671</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.1192</td>\n",
       "      <td>0.977286</td>\n",
       "      <td>0.852</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>1</td>\n",
       "      <td>S263P</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.992</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.25831</td>\n",
       "      <td>29.2</td>\n",
       "      <td>5.211912</td>\n",
       "      <td>0.980945</td>\n",
       "      <td>0.983</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.951397</td>\n",
       "      <td>0.844</td>\n",
       "      <td>0.71898</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.9852</td>\n",
       "      <td>0.860283</td>\n",
       "      <td>0.458</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>1</td>\n",
       "      <td>A167V</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.969</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.571933</td>\n",
       "      <td>34.0</td>\n",
       "      <td>5.972388</td>\n",
       "      <td>0.998261</td>\n",
       "      <td>0.980</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.853707</td>\n",
       "      <td>0.92</td>\n",
       "      <td>0.760144</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.8614</td>\n",
       "      <td>0.918544</td>\n",
       "      <td>0.766</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>1</td>\n",
       "      <td>H160R</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.952</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.553083</td>\n",
       "      <td>27.5</td>\n",
       "      <td>4.924293</td>\n",
       "      <td>0.99862</td>\n",
       "      <td>0.985</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.963709</td>\n",
       "      <td>0.919</td>\n",
       "      <td>0.675616</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.9973</td>\n",
       "      <td>0.828964</td>\n",
       "      <td>0.536</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>1</td>\n",
       "      <td>R155H</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.980</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.449379</td>\n",
       "      <td>29.4</td>\n",
       "      <td>5.229742</td>\n",
       "      <td>0.99574</td>\n",
       "      <td>0.898</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.717298</td>\n",
       "      <td>0.909</td>\n",
       "      <td>0.770769</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.6977</td>\n",
       "      <td>0.897079</td>\n",
       "      <td>0.647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  GeneSymbol  ClinSigSimple Variant UniprotID  SIFT_label  SIFT_score  PolyPhen_label  PolyPhen_score  BayesDel_label BayesDel_score  CADD_PHRED_score  CADD_RAW_score ClinPred_score  VEST4_score  EVE_label EVE_score REVEL_score PrimateAI_score  AM_label AM_score  Envision_predictions  QAFI_predictions MetaRNN_score  MetaRNN_pred\n",
       "0       AAAS              0   G533R    Q9NRG9         1.0        0.00             NaN           0.000             0.0      -0.362112              20.9        2.951810       0.062144        0.420        NaN       NaN       0.257         0.52671       0.0   0.1192              0.977286             0.852           NaN           NaN\n",
       "1       AAAS              1   S263P    Q9NRG9         1.0        0.00             1.0           0.992             1.0        0.25831              29.2        5.211912       0.980945        0.983        1.0  0.951397       0.844         0.71898       1.0   0.9852              0.860283             0.458           NaN           NaN\n",
       "2       AAAS              1   A167V    Q9NRG9         1.0        0.03             1.0           0.969             1.0       0.571933              34.0        5.972388       0.998261        0.980        1.0  0.853707        0.92        0.760144       1.0   0.8614              0.918544             0.766           NaN           NaN\n",
       "3       AAAS              1   H160R    Q9NRG9         1.0        0.00             1.0           0.952             1.0       0.553083              27.5        4.924293        0.99862        0.985        1.0  0.963709       0.919        0.675616       1.0   0.9973              0.828964             0.536           NaN           NaN\n",
       "4       AAAS              1   R155H    Q9NRG9         1.0        0.02             1.0           0.980             1.0       0.449379              29.4        5.229742        0.99574        0.898        1.0  0.717298       0.909        0.770769       1.0   0.6977              0.897079             0.647           NaN           NaN"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "f34be4ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "# load extra QAFI scores, align column names\n",
    "new = (pd.read_csv(\"clinvarall.csv\")\n",
    "         .rename(columns={'uniprot': 'UniprotID',\n",
    "                          'variant': 'Variant',\n",
    "                          'qafi1': 'QAFI_predictions'}))\n",
    "\n",
    "# QAFI convention: ↑ benign, ↓ pathogenic  →  multiply by –1 so ↑ pathogenic\n",
    "new['QAFI_predictions'] = -new['QAFI_predictions']\n",
    "\n",
    "# merge and fill gaps only\n",
    "combined = combined.merge(new[['UniprotID', 'Variant', 'QAFI_predictions']],\n",
    "                          on=['UniprotID', 'Variant'],\n",
    "                          how='left',\n",
    "                          suffixes=('', '_new'))\n",
    "\n",
    "combined['QAFI_predictions'] = combined['QAFI_predictions'].fillna(\n",
    "                                   combined['QAFI_predictions_new'])\n",
    "combined.drop(columns='QAFI_predictions_new', inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "id": "a0e40814",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GeneSymbol                  0\n",
       "ClinSigSimple               0\n",
       "Variant                     0\n",
       "UniprotID                   0\n",
       "SIFT_label               1099\n",
       "SIFT_score               1099\n",
       "PolyPhen_label           6891\n",
       "PolyPhen_score           3330\n",
       "BayesDel_label             69\n",
       "BayesDel_score             55\n",
       "CADD_PHRED_score            0\n",
       "CADD_RAW_score              0\n",
       "ClinPred_score            141\n",
       "VEST4_score              1108\n",
       "EVE_label               22860\n",
       "EVE_score               15768\n",
       "REVEL_score              3876\n",
       "PrimateAI_score           510\n",
       "AM_label                 6958\n",
       "AM_score                 2883\n",
       "Envision_predictions     3583\n",
       "QAFI_predictions         4687\n",
       "MetaRNN_score           37287\n",
       "MetaRNN_pred            37396\n",
       "dtype: int64"
      ]
     },
     "execution_count": 140,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined.isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "71963a78",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "min    0.0\n",
       "max    1.0\n",
       "Name: VEST4_score, dtype: float64"
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined['VEST4_score'].agg(['min', 'max'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "793ce951",
   "metadata": {},
   "source": [
    "Label 0 has higher (more damaging) scores: 0 = pathogenic, 1 = benign; in **QAFI** and **Envision**.\n",
    "\n",
    "Thus, we must flip the labels and rerun."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "id": "3c5b62a4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "min    0.1540\n",
       "max    1.5025\n",
       "Name: QAFI_predictions, dtype: float64"
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined['QAFI_predictions'].agg(['min', 'max'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "8a032147",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "min    0.421637\n",
       "max    1.127760\n",
       "Name: Envision_predictions, dtype: float64"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined['Envision_predictions'].agg(['min', 'max'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "56e17f6a",
   "metadata": {},
   "outputs": [],
   "source": [
    "combined['QAFI_predictions']   *= -1\n",
    "combined['Envision_predictions'] *= -1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "id": "79a5cb04",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "min   -1.5025\n",
       "max   -0.1540\n",
       "Name: QAFI_predictions, dtype: float64"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined['QAFI_predictions'].agg(['min', 'max'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "id": "f12f4f23",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "min   -1.127760\n",
       "max   -0.421637\n",
       "Name: Envision_predictions, dtype: float64"
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined['Envision_predictions'].agg(['min', 'max'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "8f3e8cc2",
   "metadata": {},
   "outputs": [],
   "source": [
    "combined.to_csv('~/Desktop/analysis files/ClinHum_ALL_preds.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc207341",
   "metadata": {},
   "source": [
    "### Table: Percentage of missing predictions for all tools for the combined Clinvar and Humsavar dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e626256c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/cc/jxs8yxmn00bb_43d9z_0f13c0000gn/T/ipykernel_19535/2962565520.py:1: DtypeWarning: Columns (12,17) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  combined = pd.read_csv('~/Desktop/analysis files/ClinHum_ALL_preds_cleaned.csv', sep=',')\n"
     ]
    }
   ],
   "source": [
    "combined = pd.read_csv('~/Desktop/analysis files/ClinHum_ALL_preds_cleaned.csv', sep=',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "a4f03481",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['GeneSymbol', 'ClinSigSimple', 'variant', 'uniprot', 'SIFT_label',\n",
       "       'SIFT_score', 'PolyPhen_label', 'PolyPhen_score', 'BayesDel (binary)',\n",
       "       'BayesDel', 'CADD', 'CADD (RAW)', 'ClinPred_score', 'VEST4',\n",
       "       'EVE (binary)', 'EVE', 'REVEL', 'PrimateAI_score',\n",
       "       'AlphaMissense (binary)', 'AlphaMissense', 'Envision', 'QAFI',\n",
       "       'MetaRNN', 'MetaRNN (binary)'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "ccfa814a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>variant</th>\n",
       "      <th>uniprot</th>\n",
       "      <th>SIFT_label</th>\n",
       "      <th>SIFT_score</th>\n",
       "      <th>PolyPhen_label</th>\n",
       "      <th>PolyPhen_score</th>\n",
       "      <th>BayesDel (binary)</th>\n",
       "      <th>BayesDel</th>\n",
       "      <th>CADD</th>\n",
       "      <th>CADD (RAW)</th>\n",
       "      <th>ClinPred_score</th>\n",
       "      <th>VEST4</th>\n",
       "      <th>EVE (binary)</th>\n",
       "      <th>EVE</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>PrimateAI_score</th>\n",
       "      <th>AlphaMissense (binary)</th>\n",
       "      <th>AlphaMissense</th>\n",
       "      <th>Envision</th>\n",
       "      <th>QAFI</th>\n",
       "      <th>MetaRNN</th>\n",
       "      <th>MetaRNN (binary)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>0</td>\n",
       "      <td>G533R</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.362112</td>\n",
       "      <td>20.9</td>\n",
       "      <td>2.951810</td>\n",
       "      <td>0.062144</td>\n",
       "      <td>0.420</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.257</td>\n",
       "      <td>0.52671</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.1192</td>\n",
       "      <td>-0.977286</td>\n",
       "      <td>-0.852</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>1</td>\n",
       "      <td>S263P</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.992</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.258310</td>\n",
       "      <td>29.2</td>\n",
       "      <td>5.211912</td>\n",
       "      <td>0.980945</td>\n",
       "      <td>0.983</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.951397</td>\n",
       "      <td>0.844</td>\n",
       "      <td>0.71898</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.9852</td>\n",
       "      <td>-0.860283</td>\n",
       "      <td>-0.458</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>1</td>\n",
       "      <td>A167V</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.969</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.571933</td>\n",
       "      <td>34.0</td>\n",
       "      <td>5.972388</td>\n",
       "      <td>0.998261</td>\n",
       "      <td>0.980</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.853707</td>\n",
       "      <td>0.920</td>\n",
       "      <td>0.760144</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.8614</td>\n",
       "      <td>-0.918544</td>\n",
       "      <td>-0.766</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>1</td>\n",
       "      <td>H160R</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.952</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.553083</td>\n",
       "      <td>27.5</td>\n",
       "      <td>4.924293</td>\n",
       "      <td>0.99862</td>\n",
       "      <td>0.985</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.963709</td>\n",
       "      <td>0.919</td>\n",
       "      <td>0.675616</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.9973</td>\n",
       "      <td>-0.828964</td>\n",
       "      <td>-0.536</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>1</td>\n",
       "      <td>R155H</td>\n",
       "      <td>Q9NRG9</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.980</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.449379</td>\n",
       "      <td>29.4</td>\n",
       "      <td>5.229742</td>\n",
       "      <td>0.99574</td>\n",
       "      <td>0.898</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.717298</td>\n",
       "      <td>0.909</td>\n",
       "      <td>0.770769</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.6977</td>\n",
       "      <td>-0.897079</td>\n",
       "      <td>-0.647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  GeneSymbol  ClinSigSimple variant uniprot  SIFT_label  SIFT_score  PolyPhen_label  PolyPhen_score  BayesDel (binary)  BayesDel  CADD  CADD (RAW) ClinPred_score  VEST4  EVE (binary)       EVE  REVEL PrimateAI_score  AlphaMissense (binary)  AlphaMissense  Envision   QAFI  MetaRNN  MetaRNN (binary)\n",
       "0       AAAS              0   G533R  Q9NRG9         1.0        0.00             NaN           0.000                0.0 -0.362112  20.9    2.951810       0.062144  0.420           NaN       NaN  0.257         0.52671                     0.0         0.1192 -0.977286 -0.852      NaN               NaN\n",
       "1       AAAS              1   S263P  Q9NRG9         1.0        0.00             1.0           0.992                1.0  0.258310  29.2    5.211912       0.980945  0.983           1.0  0.951397  0.844         0.71898                     1.0         0.9852 -0.860283 -0.458      NaN               NaN\n",
       "2       AAAS              1   A167V  Q9NRG9         1.0        0.03             1.0           0.969                1.0  0.571933  34.0    5.972388       0.998261  0.980           1.0  0.853707  0.920        0.760144                     1.0         0.8614 -0.918544 -0.766      NaN               NaN\n",
       "3       AAAS              1   H160R  Q9NRG9         1.0        0.00             1.0           0.952                1.0  0.553083  27.5    4.924293        0.99862  0.985           1.0  0.963709  0.919        0.675616                     1.0         0.9973 -0.828964 -0.536      NaN               NaN\n",
       "4       AAAS              1   R155H  Q9NRG9         1.0        0.02             1.0           0.980                1.0  0.449379  29.4    5.229742        0.99574  0.898           1.0  0.717298  0.909        0.770769                     1.0         0.6977 -0.897079 -0.647      NaN               NaN"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "666e5dcd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GeneSymbol                    0\n",
       "ClinSigSimple                 0\n",
       "variant                       0\n",
       "uniprot                       0\n",
       "SIFT_label                 1099\n",
       "SIFT_score                 1099\n",
       "PolyPhen_label             6891\n",
       "PolyPhen_score             3330\n",
       "BayesDel (binary)            69\n",
       "BayesDel                     69\n",
       "CADD                          0\n",
       "CADD (RAW)                    0\n",
       "ClinPred_score              141\n",
       "VEST4                      1108\n",
       "EVE (binary)              22860\n",
       "EVE                       16735\n",
       "REVEL                      3975\n",
       "PrimateAI_score             510\n",
       "AlphaMissense (binary)     6958\n",
       "AlphaMissense              3202\n",
       "Envision                   3583\n",
       "QAFI                       4687\n",
       "MetaRNN                   37396\n",
       "MetaRNN (binary)          37396\n",
       "dtype: int64"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined.isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "98c46211",
   "metadata": {},
   "outputs": [],
   "source": [
    "score_cols  = ['CADD','MetaRNN','Envision','QAFI','EVE','AlphaMissense','BayesDel','REVEL','VEST4']\n",
    "\n",
    "# % of variants with no preds for each predictor\n",
    "missing_df = (\n",
    "    combined[score_cols].isna().mean()\n",
    "             .mul(100).round(1)\n",
    "             .rename('ClinVar + Humsavar')\n",
    "             .reset_index()\n",
    "             .rename(columns={'index':'Predictor'})\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "24dc9337",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Predictor</th>\n",
       "      <th>ClinVar + Humsavar</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CADD</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MetaRNN</td>\n",
       "      <td>79.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Envision</td>\n",
       "      <td>7.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>QAFI</td>\n",
       "      <td>9.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EVE</td>\n",
       "      <td>35.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AlphaMissense</td>\n",
       "      <td>6.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>BayesDel</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>REVEL</td>\n",
       "      <td>8.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>VEST4</td>\n",
       "      <td>2.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Predictor  ClinVar + Humsavar\n",
       "0           CADD                 0.0\n",
       "1        MetaRNN                79.3\n",
       "2       Envision                 7.6\n",
       "3           QAFI                 9.9\n",
       "4            EVE                35.5\n",
       "5  AlphaMissense                 6.8\n",
       "6       BayesDel                 0.1\n",
       "7          REVEL                 8.4\n",
       "8          VEST4                 2.3"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missing_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "dd741a0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "missing_df.to_latex(\"missing_preds_table_PreAnalysis.tex\", index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "oasis",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
